Index: fsl-rc-local/rc.local.etc
===================================================================
--- fsl-rc-local/rc.local.etc
+++ fsl-rc-local/rc.local.etc
@@ -30,6 +30,13 @@ if [ -e /usr/bin/ec_uevent ];then
         ec_uevent &
 fi
 
+if [ ! -e /cache/ota ];then
+        mkdir /cache/ota
+fi
+if [ ! -e /cache/ota/agent_config.xml ];then
+        cp /usr/local/OTA-Agent/agent_config.xml /cache/ota
+fi
+
 
 exit 0
 
