From 6e420ec2cd3117259540637f0fa16027293ced92 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Tue, 19 Jan 2016 15:51:45 +0800
Subject: [PATCH] Kconfig


diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 0c82b17..ee3d242 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -523,6 +523,71 @@ config TARGET_MX6SABRESD
 	select CPU_V7
 	select SUPPORT_SPL
 
+config TARGET_MX6RSB4410A1_1G
+	bool "Support mx6rsb4410a1 1G"
+	select CPU_V7
+	select SUPPORT_SPL
+
+config TARGET_MX6RSB4410A2_1G
+	bool "Support mx6rsb4410a2 1G"
+	select CPU_V7
+	select SUPPORT_SPL
+
+config TARGET_MX6UBCDS31A1_1G
+	bool "Support mx6ubcds31a1 1G"
+	select CPU_V7
+	select SUPPORT_SPL
+
+config TARGET_MX6UBC220A1_1G
+	bool "Support mx6ub220a1 1G"
+	select CPU_V7
+	select SUPPORT_SPL
+
+config TARGET_MX6ROM3420A1_1G
+	bool "Support mx6rom3420a1 1G"
+	select CPU_V7
+	select SUPPORT_SPL
+
+config TARGET_MX6ROM3420A1_2G
+	bool "Support mx6rom3420a1 2G"
+	select CPU_V7
+	select SUPPORT_SPL
+
+config TARGET_MX6ROM5420A1_1G
+	bool "Support mx6rom5420a1 1G"
+	select CPU_V7
+	select SUPPORT_SPL
+
+config TARGET_MX6ROM5420A1_2G
+	bool "Support mx6rom5420a1 2G"
+	select CPU_V7
+	select SUPPORT_SPL
+
+config TARGET_MX6ROM5420B1_1G
+	bool "Support mx6rom5420b1 1G"
+	select CPU_V7
+	select SUPPORT_SPL
+
+config TARGET_MX6ROM5420B1_2G
+	bool "Support mx6rom5420b1 2G"
+	select CPU_V7
+	select SUPPORT_SPL
+
+config TARGET_MX6ROM7420A1_1G
+	bool "Support mx6rom7420a1 1G"
+	select CPU_V7
+	select SUPPORT_SPL
+
+config TARGET_MX6ROM7421A1_1G
+	bool "Support mx6rom7421a1 1G"
+	select CPU_V7
+	select SUPPORT_SPL
+
+config TARGET_MX6WISE3310A1_1G
+	bool "Support mx6romwise3310a1 1G"
+	select CPU_V7
+	select SUPPORT_SPL
+
 config TARGET_MX6SLEVK
 	bool "Support mx6slevk"
 	select CPU_V7
@@ -867,6 +932,7 @@ source "board/freescale/mx53smd/Kconfig"
 source "board/freescale/mx6qarm2/Kconfig"
 source "board/freescale/mx6qsabreauto/Kconfig"
 source "board/freescale/mx6sabresd/Kconfig"
+source "board/freescale/mx6advantech/Kconfig"
 source "board/freescale/mx6slevk/Kconfig"
 source "board/freescale/mx6sxsabresd/Kconfig"
 source "board/freescale/mx6sxsabreauto/Kconfig"
-- 
1.7.9.5

