From bf037d6b81211b887a466fdc9ff7e02013c0623a Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Tue, 19 Jan 2016 16:08:49 +0800
Subject: [PATCH] uboot_crc_makefile


diff --git a/Makefile b/Makefile
index 950c960..ee349d6 100644
--- a/Makefile
+++ b/Makefile
@@ -726,7 +726,7 @@ DO_STATIC_RELA =
 endif
 
 # Always append ALL so that arch config.mk's can add custom ones
-ALL-y += u-boot.srec u-boot.bin System.map binary_size_check
+ALL-y += u-boot.srec u-boot.bin u-boot_crc.bin System.map binary_size_check
 
 ALL-$(CONFIG_ONENAND_U_BOOT) += u-boot-onenand.bin
 ifeq ($(CONFIG_SPL_FSL_PBL),y)
@@ -848,6 +848,12 @@ u-boot.bin: u-boot FORCE
 	$(call DO_STATIC_RELA,$<,$@,$(CONFIG_SYS_TEXT_BASE))
 	$(BOARD_SIZE_CHECK)
 
+u-boot_crc.bin:	u-boot.bin
+		@cp tools/mk_uboot_crc .
+		@./mk_uboot_crc
+		@crc32 ./u-boot_crc.bin > ./u-boot_crc.bin.crc
+		@rm -f ./mk_uboot_crc
+
 u-boot.ldr:	u-boot
 		$(CREATE_LDR_ENV)
 		$(LDR) -T $(CONFIG_CPU) -c $@ $< $(LDR_FLAGS)
-- 
1.7.9.5

