From 09f47008ed7f507919418072dffb07cb19c16d54 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Tue, 19 Jan 2016 16:10:00 +0800
Subject: [PATCH] uboot_crc_tools


diff --git a/tools/Makefile b/tools/Makefile
index 4bbb153..c3a4786 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -54,7 +54,7 @@ mkenvimage-objs := mkenvimage.o os_support.o lib/crc32.o
 
 hostprogs-y += dumpimage mkimage
 hostprogs-$(CONFIG_FIT_SIGNATURE) += fit_info fit_check_sign
-
+hostprogs-y += mk_uboot_crc
 FIT_SIG_OBJS-$(CONFIG_FIT_SIGNATURE) := common/image-sig.o
 # Flattened device tree objects
 LIBFDT_OBJS := $(addprefix lib/libfdt/, \
@@ -101,7 +101,7 @@ dumpimage-objs := $(dumpimage-mkimage-objs) dumpimage.o
 mkimage-objs   := $(dumpimage-mkimage-objs) mkimage.o
 fit_info-objs   := $(dumpimage-mkimage-objs) fit_info.o
 fit_check_sign-objs   := $(dumpimage-mkimage-objs) fit_check_sign.o
-
+mk_uboot_crc-objs := mk_uboot_crc.o
 # TODO(sjg@chromium.org): Is this correct on Mac OS?
 
 ifneq ($(CONFIG_MX23)$(CONFIG_MX28),)
diff --git a/tools/mk_uboot_crc.c b/tools/mk_uboot_crc.c
new file mode 100644
index 0000000..3daa450
--- /dev/null
+++ b/tools/mk_uboot_crc.c
@@ -0,0 +1,47 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#include <errno.h>
+
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/ioctl.h>
+#include <sys/stat.h>
+#include <arpa/inet.h>
+#include <fcntl.h>
+#include <sys/wait.h>
+
+#define IMG_SZ 614400
+
+int  main(void)
+{
+	struct stat buf;
+	unsigned char binary[IMG_SZ];
+	int fd;
+	memset(binary, 0, IMG_SZ);
+	
+	if(stat("./u-boot.bin", &buf) != 0)
+		return -1;
+	//printf("File size %d\n", buf.st_size);
+	
+	fd = open("./u-boot.bin", O_RDWR);
+	if(fd < 0)
+	{
+		printf("Open u-boot.bin file fail %d\n", fd);
+		return -1;
+	}
+	read(fd, binary, buf.st_size);
+	close(fd);
+	
+	fd = open("./u-boot_crc.bin", O_RDWR | O_CREAT, S_IRWXU);
+	if(fd < 0)
+	{
+		printf("Open file fail %d\n", fd);
+		return -1;
+	}
+	write(fd, binary, IMG_SZ);
+
+	//printf("padding done\n");
+	return 0;
+} 
-- 
1.7.9.5

