From 12915b1d8b6b8d2233365cb76c0a0fcb6fd10d70 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Wed, 6 Jan 2016 17:16:42 +0800
Subject: [PATCH] common/board_r.c


diff --git a/common/board_r.c b/common/board_r.c
old mode 100644
new mode 100755
index b9f8aeb..1d5059c
--- a/common/board_r.c
+++ b/common/board_r.c
@@ -58,6 +58,21 @@
 #ifdef CONFIG_FASTBOOT
 #include <fastboot.h>
 #endif
+#ifdef CONFIG_ADVANTECH
+#include <version.h>
+#include <spi_flash.h>
+
+#define CONFIG_SPI_ENV_OFFSET	(768 * 1024)
+#define CONFIG_SPI_ENV_SIZE	(8 * 1024)
+
+struct boardcfg_t {
+    unsigned char mac[6];
+    unsigned char sn[10];
+    unsigned char Manufacturing_Time[14];
+};
+
+static struct spi_flash *flash;
+#endif /* CONFIG_ADVANTECH */
 
 DECLARE_GLOBAL_DATA_PTR;
 
@@ -567,6 +582,74 @@ static int initr_ethaddr(void)
 #endif
 	return 0;
 }
+
+#ifdef CONFIG_ADVANTECH
+#define XMK_STR(x)	#x
+#define MK_STR(x)	XMK_STR(x)
+int boardcfg_get_mac(void)
+{
+	int rc = 0;
+	struct boardcfg_t boardcfg;
+	char print_buf[32];
+	uint64_t macaddr = 0;
+
+	flash = spi_flash_probe(CONFIG_SF_DEFAULT_BUS, CONFIG_SF_DEFAULT_CS,
+				CONFIG_SF_DEFAULT_SPEED, CONFIG_SF_DEFAULT_MODE);
+	if (!flash)
+		return -1;
+
+	if(spi_flash_read(flash, CONFIG_SPI_ENV_OFFSET + 8*CONFIG_SPI_ENV_SIZE, sizeof(boardcfg), &boardcfg)==0) {
+
+		/*printf("offset=%d\n", CONFIG_SPI_ENV_OFFSET+ 8*CONFIG_SPI_ENV_SIZE);*/
+
+		/*printf("0x%02X:0x%02X:0x%02X:0x%02X:0x%02X:0x%02X\n",
+						boardcfg.mac[0],
+						boardcfg.mac[1],
+						boardcfg.mac[2],
+						boardcfg.mac[3],
+						boardcfg.mac[4],
+						boardcfg.mac[5]);*/
+
+
+		macaddr = ((uint64_t)boardcfg.mac[0] << 40)
+			+ ((uint64_t)boardcfg.mac[1] << 32)
+			+ ((uint64_t)boardcfg.mac[2] << 24)
+			+ ((uint64_t)boardcfg.mac[3] << 16)
+			+ ((uint64_t)boardcfg.mac[4] << 8)
+			+ boardcfg.mac[5];
+		/* printf ("MAC addr =%012llX\n", macaddr); */
+
+		if( (macaddr==0) || (macaddr==0xFFFFFFFFFFFFull) ) {
+			printf("MAC address is invailed !!\n");
+			sprintf(print_buf,"0x00:0x04:0x9F:0x01:0x30:0xE0");
+			printf("Use default MAC adderss:%s\n",print_buf);
+			setenv("ethaddr",print_buf);
+			return rc;
+		}
+	} else {
+		printf("SPI Read fail!!\n");
+		rc = -1;
+	}
+	
+	if (rc==0) {
+		sprintf(print_buf, "0x%02X:0x%02X:0x%02X:0x%02X:0x%02X:0x%02X", 
+						boardcfg.mac[0],
+						boardcfg.mac[1],
+						boardcfg.mac[2],
+						boardcfg.mac[3],
+						boardcfg.mac[4],
+						boardcfg.mac[5]);
+		printf ("MAC addr = %s\n", print_buf);
+
+		if( (strcmp (getenv("ethaddr"),print_buf) != 0)||(getenv("ethaddr") == NULL) 
+			|| (strcmp (getenv("ethaddr"),MK_STR(CONFIG_ETHADDR)) == 0) ) {
+			setenv("ethaddr", print_buf);
+		}
+	}
+	
+	return rc;
+}
+#endif /* CONFIG_ADVANTECH */
 #endif /* CONFIG_CMD_NET */
 
 #ifdef CONFIG_CMD_KGDB
@@ -898,6 +981,9 @@ init_fnc_t init_sequence_r[] = {
 	/* PPC has a udelay(20) here dating from 2002. Why? */
 #ifdef CONFIG_CMD_NET
 	initr_ethaddr,
+#ifdef CONFIG_ADVANTECH
+	boardcfg_get_mac,	/* Get MAC address from SPI */
+#endif
 #endif
 #ifdef CONFIG_BOARD_LATE_INIT
 	board_late_init,
-- 
1.7.9.5

