From 484565ddcc26302042079a8512ea3a28ccbcf11c Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Tue, 19 Jan 2016 16:26:56 +0800
Subject: [PATCH] 0006_spi_002_u-boot_environment_storge_in_spi


diff --git a/arch/arm/include/asm/arch-mx6/imx-regs.h b/arch/arm/include/asm/arch-mx6/imx-regs.h
index 9fbe3e4..72a9374 100644
--- a/arch/arm/include/asm/arch-mx6/imx-regs.h
+++ b/arch/arm/include/asm/arch-mx6/imx-regs.h
@@ -654,7 +654,11 @@ struct cspi_regs {
 #define MXC_CSPICTRL_TC		(1 << 7)
 #define MXC_CSPICTRL_RXOVF	(1 << 6)
 #define MXC_CSPIPERIOD_32KHZ	(1 << 15)
-#define MAX_SPI_BYTES	32
+#ifdef CONFIG_ADVANTECH
+#define MAX_SPI_BYTES   (64 * 4)
+#else
+#define MAX_SPI_BYTES   32
+#endif
 #if defined(CONFIG_MX6SL) || defined(CONFIG_MX6DL) || defined(CONFIG_MX6UL)
 #define SPI_MAX_NUM	3
 #else
-- 
1.7.9.5

