From 0c9a8739cf612b38db2d3b3d7151a148e3f654db Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Wed, 20 Jan 2016 11:07:28 +0800
Subject: [PATCH] 0008_spi_001_use_index_1_for_SD_3_for_emmc


diff --git a/board/freescale/mx6advantech/mx6advantech.c b/board/freescale/mx6advantech/mx6advantech.c
index a4e87b4..69ce6a4 100644
--- a/board/freescale/mx6advantech/mx6advantech.c
+++ b/board/freescale/mx6advantech/mx6advantech.c
@@ -428,8 +428,14 @@ int board_mmc_init(bd_t *bis)
 			       i + 1, CONFIG_SYS_FSL_USDHC_NUM);
 			return -EINVAL;
 		}
-
+#ifdef CONFIG_ADVANTECH
+		/* We use index 1 for SD card and index 3 for eMMC */
+		if (i != 1) {
+			ret = fsl_esdhc_initialize(bis, &usdhc_cfg[i]);
+			}
+#else
 		ret = fsl_esdhc_initialize(bis, &usdhc_cfg[i]);
+#endif
 		if (ret)
 			return ret;
 	}
-- 
1.7.9.5

