From 501598d185ad98f9432c69722328a2a1b83b1433 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Tue, 5 Jan 2016 15:05:59 +0800
Subject: [PATCH] svn up


diff --git a/arch/arm/imx-common/sata.c b/arch/arm/imx-common/sata.c
index 2e69486..b3c4013 100644
--- a/arch/arm/imx-common/sata.c
+++ b/arch/arm/imx-common/sata.c
@@ -17,7 +17,11 @@ int setup_sata(void)
 	int ret = enable_sata_clock();
 	if (ret)
 		return ret;
-
+#ifdef CONFIG_SATA_GEN2
+	clrsetbits_le32(&iomuxc_regs->gpr[13],
+			IOMUXC_GPR13_SATA_MASK,
+			CONFIG_SATA_GEN2);
+#else
 	clrsetbits_le32(&iomuxc_regs->gpr[13],
 			IOMUXC_GPR13_SATA_MASK,
 			IOMUXC_GPR13_SATA_PHY_8_RXEQ_3P0DB
@@ -29,6 +33,6 @@ int setup_sata(void)
 			|IOMUXC_GPR13_SATA_PHY_3_TXBOOST_0P00_DB
 			|IOMUXC_GPR13_SATA_PHY_2_TX_1P104V
 			|IOMUXC_GPR13_SATA_PHY_1_SLOW);
-
+#endif
 	return 0;
 }
-- 
1.7.9.5

