From 17a5fe24c912da1b92acda2aa68ff01667e9dba9 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Wed, 20 Jan 2016 15:36:17 +0800
Subject: [PATCH] disable pmic


diff --git a/board/freescale/mx6advantech/mx6advantech.c b/board/freescale/mx6advantech/mx6advantech.c
index ca96049..f547fb0 100755
--- a/board/freescale/mx6advantech/mx6advantech.c
+++ b/board/freescale/mx6advantech/mx6advantech.c
@@ -1034,7 +1034,7 @@ int board_init(void)
 
 	return 0;
 }
-
+#ifndef CONFIG_ADVANTECH
 static struct pmic *pfuze;
 int power_init_board(void)
 {
@@ -1116,7 +1116,7 @@ int power_init_board(void)
 
 	return 0;
 }
-
+#endif
 #ifdef CONFIG_LDO_BYPASS_CHECK
 void ldo_mode_set(int ldo_bypass)
 {
diff --git a/include/configs/mx6rom3420.h b/include/configs/mx6rom3420.h
index 65deac8..a46b31f 100755
--- a/include/configs/mx6rom3420.h
+++ b/include/configs/mx6rom3420.h
@@ -93,11 +93,6 @@
 #define CONFIG_PCIE_IMX_POWER_GPIO	IMX_GPIO_NR(3, 19)
 #endif
 
-/* PMIC */
-#define CONFIG_POWER
-#define CONFIG_POWER_I2C
-#define CONFIG_POWER_PFUZE100
-#define CONFIG_POWER_PFUZE100_I2C_ADDR	0x08
 
 /* USB Configs */
 #define CONFIG_CMD_USB
diff --git a/include/configs/mx6rom5420.h b/include/configs/mx6rom5420.h
index ce0f8c7..2ae123b 100755
--- a/include/configs/mx6rom5420.h
+++ b/include/configs/mx6rom5420.h
@@ -55,6 +55,8 @@
 #define PHYS_SDRAM_SIZE         (512u * 1024 * 1024)
 #endif
 #endif
+
+#include "mx6advantech_common.h"
 /* don't use pmic */
 #undef CONFIG_LDO_BYPASS_CHECK
 
@@ -69,7 +71,6 @@
 #ifdef CONFIG_SPL_BUILD
 #define CONFIG_BOOT_SELECT
 #endif
-#include "mx6advantech_common.h"
 
 #define CONFIG_SYS_FSL_USDHC_NUM	3
 #define CONFIG_SYS_MMC_ENV_DEV		1	/* SDHC3 */
@@ -103,11 +104,6 @@
 #define CONFIG_PCIE_IMX_POWER_GPIO	IMX_GPIO_NR(3, 19)
 #endif
 
-/* PMIC */
-#define CONFIG_POWER
-#define CONFIG_POWER_I2C
-#define CONFIG_POWER_PFUZE100
-#define CONFIG_POWER_PFUZE100_I2C_ADDR	0x08
 
 /* USB Configs */
 #define CONFIG_CMD_USB
diff --git a/include/configs/mx6rom7420.h b/include/configs/mx6rom7420.h
index 195cc0e..10edc39 100755
--- a/include/configs/mx6rom7420.h
+++ b/include/configs/mx6rom7420.h
@@ -81,11 +81,6 @@
 #define CONFIG_PCIE_IMX_POWER_GPIO	IMX_GPIO_NR(3, 19)
 #endif
 
-/* PMIC */
-#define CONFIG_POWER
-#define CONFIG_POWER_I2C
-#define CONFIG_POWER_PFUZE100
-#define CONFIG_POWER_PFUZE100_I2C_ADDR	0x08
 
 /* USB Configs */
 #define CONFIG_CMD_USB
diff --git a/include/configs/mx6rsb4410.h b/include/configs/mx6rsb4410.h
index bcd59b4..b75f3dd 100755
--- a/include/configs/mx6rsb4410.h
+++ b/include/configs/mx6rsb4410.h
@@ -69,11 +69,6 @@
 #define CONFIG_PCIE_IMX_POWER_GPIO	IMX_GPIO_NR(3, 19)
 #endif
 
-/* PMIC */
-#define CONFIG_POWER
-#define CONFIG_POWER_I2C
-#define CONFIG_POWER_PFUZE100
-#define CONFIG_POWER_PFUZE100_I2C_ADDR	0x08
 
 /* USB Configs */
 #define CONFIG_CMD_USB
diff --git a/include/configs/mx6ubc220.h b/include/configs/mx6ubc220.h
index 69d3099..a354f6d 100755
--- a/include/configs/mx6ubc220.h
+++ b/include/configs/mx6ubc220.h
@@ -35,10 +35,11 @@
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-ubc220-a1.dtb"
 #define PHYS_SDRAM_SIZE		(512u * 1024 * 1024)
 #endif
+
+#include "mx6advantech_common.h"
 #define CONFIG_MMC_CLOCK_DOWNGRADE
 /* don't use pmic */
 #undef CONFIG_LDO_BYPASS_CHECK
-#include "mx6advantech_common.h"
 
 #define CONFIG_SYS_FSL_USDHC_NUM	3
 #define CONFIG_SYS_MMC_ENV_DEV		1	/* SDHC3 */
@@ -73,11 +74,6 @@
 #define CONFIG_PCIE_IMX_POWER_GPIO	IMX_GPIO_NR(3, 19)
 #endif
 
-/* PMIC */
-#define CONFIG_POWER
-#define CONFIG_POWER_I2C
-#define CONFIG_POWER_PFUZE100
-#define CONFIG_POWER_PFUZE100_I2C_ADDR	0x08
 
 /* USB Configs */
 #define CONFIG_CMD_USB
diff --git a/include/configs/mx6ubcds31.h b/include/configs/mx6ubcds31.h
index 2b1b8a4..93f3033 100755
--- a/include/configs/mx6ubcds31.h
+++ b/include/configs/mx6ubcds31.h
@@ -69,11 +69,6 @@
 #define CONFIG_PCIE_IMX_POWER_GPIO	IMX_GPIO_NR(3, 19)
 #endif
 
-/* PMIC */
-#define CONFIG_POWER
-#define CONFIG_POWER_I2C
-#define CONFIG_POWER_PFUZE100
-#define CONFIG_POWER_PFUZE100_I2C_ADDR	0x08
 
 /* USB Configs */
 #define CONFIG_CMD_USB
diff --git a/include/configs/mx6wise3310.h b/include/configs/mx6wise3310.h
index c5113c6..58c1c85 100755
--- a/include/configs/mx6wise3310.h
+++ b/include/configs/mx6wise3310.h
@@ -69,11 +69,6 @@
 #define CONFIG_PCIE_IMX_POWER_GPIO	IMX_GPIO_NR(3, 19)
 #endif
 
-/* PMIC */
-#define CONFIG_POWER
-#define CONFIG_POWER_I2C
-#define CONFIG_POWER_PFUZE100
-#define CONFIG_POWER_PFUZE100_I2C_ADDR	0x08
 
 /* USB Configs */
 #define CONFIG_CMD_USB
-- 
1.7.9.5

