From 21f9b78bcf54448ab06f98dcd854348f61f675c7 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Tue, 26 Jan 2016 15:09:24 +0800
Subject: [PATCH] te


diff --git a/include/configs/mx6rom7421.h b/include/configs/mx6rom7421.h
old mode 100644
new mode 100755
index ff7f6f1..219cc6c
--- a/include/configs/mx6rom7421.h
+++ b/include/configs/mx6rom7421.h
@@ -22,6 +22,10 @@
 #define CONFIG_MXC_UART_BASE	UART1_BASE
 #define CONFIG_CONSOLE_DEV		"ttymxc0"
 #define CONFIG_MMCROOT			"/dev/mmcblk0p2"  /* SDHC3 */
+
+/* support SATA boot */
+#define CONFIG_SATA_BOOT
+#define CONFIG_SATA_GEN2        0x0593e4c4
 #if defined(CONFIG_MX6QP)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6qp-rom7421-a1.dtb"
 #define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
@@ -37,6 +41,8 @@
 #endif
 
 #include "mx6advantech_common.h"
+/* don't use pmic */
+#undef CONFIG_LDO_BYPASS_CHECK
 
 #define CONFIG_SYS_FSL_USDHC_NUM	3
 #define CONFIG_SYS_MMC_ENV_DEV		1	/* SDHC3 */
@@ -46,6 +52,12 @@
 #define CONFIG_SF_DEFAULT_CS   0
 #endif
 
+#ifdef CONFIG_CMD_SF
+#ifdef CONFIG_SPI_FLASH_CS
+#undef CONFIG_SPI_FLASH_CS
+#define CONFIG_SPI_FLASH_CS	1
+#endif
+#endif
 /*
  * imx6 q/dl/solo pcie would be failed to work properly in kernel, if
  * the pcie module is iniialized/enumerated both in uboot and linux
@@ -67,11 +79,6 @@
 #define CONFIG_PCIE_IMX_POWER_GPIO	IMX_GPIO_NR(3, 19)
 #endif
 
-/* PMIC */
-#define CONFIG_POWER
-#define CONFIG_POWER_I2C
-#define CONFIG_POWER_PFUZE100
-#define CONFIG_POWER_PFUZE100_I2C_ADDR	0x08
 
 /* USB Configs */
 #define CONFIG_CMD_USB
@@ -120,12 +127,12 @@
 #define LCD_VDD_EN 		IMX_GPIO_NR(6, 7)	
 #endif
 
-#define SPI1_CS0                IMX_GPIO_NR(2,30)
+#define SPI1_CS0                IMX_GPIO_NR(3,19)
 #define IOMUX_SPI_SCLK          MX6_PAD_EIM_D16__ECSPI1_SCLK
 #define IOMUX_SPI_MISO          MX6_PAD_EIM_D17__ECSPI1_MISO
 #define IOMUX_SPI_MOSI          MX6_PAD_EIM_D18__ECSPI1_MOSI
-#define IOMUX_SPI_CS0           MX6_PAD_EIM_EB2__ECSPI1_SS0
+#define IOMUX_SPI_CS0           MX6_PAD_EIM_D19__ECSPI1_SS1
 
-#define USDHC2_CD_GPIO          IMX_GPIO_NR(2, 2)
-#define USDHC3_CD_GPIO          IMX_GPIO_NR(2, 0)
+#define USDHC2_CD_GPIO          IMX_GPIO_NR(2, 0)
+#define USDHC3_CD_GPIO          IMX_GPIO_NR(2, 1)
 #endif                         /* __MX6QSABRESD_CONFIG_H */
-- 
1.7.9.5

