From 84a841acca94db9dbeb79e00468db376df4044b1 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Wed, 20 Jan 2016 17:10:12 +0800
Subject: [PATCH] lvds si


diff --git a/board/freescale/mx6advantech/mx6advantech.c b/board/freescale/mx6advantech/mx6advantech.c
index f547fb0..e5de79b 100755
--- a/board/freescale/mx6advantech/mx6advantech.c
+++ b/board/freescale/mx6advantech/mx6advantech.c
@@ -998,14 +998,6 @@ void setup_lvds_init(void)
 	gpio_direction_output(LCD_VDD_EN, 0);
 }
 
-void setup_lvds_poweron(void)
-{
-	//LCD_BKLT_EN
-	gpio_direction_output(LCD_BKLT_EN, 1);
-
-	//LCD_VDD_EN
-	gpio_direction_output(LCD_VDD_EN, 1);
-}
 #endif
 int board_init(void)
 {
@@ -1013,7 +1005,6 @@ int board_init(void)
 	gd->bd->bi_boot_params = PHYS_SDRAM + 0x100;
 #if defined (CONFIG_ADVANTECH) && defined(CONFIG_SUPPORT_LVDS)
        setup_lvds_init();
-       setup_lvds_poweron();
 #endif
 #ifdef CONFIG_MXC_SPI
 	setup_spi();
-- 
1.7.9.5

