From 04a71c47029980ea37164ebea75cd559c151bdeb Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 21 Jan 2016 17:02:11 +0800
Subject: [PATCH] twosd2


diff --git a/common/board_r.c b/common/board_r.c
index a455e30..57b0e93 100644
--- a/common/board_r.c
+++ b/common/board_r.c
@@ -860,6 +860,15 @@ int board_set_boot_device(void)
 			if(emmc_exist) 	setenv("mmcdev", "1");
 			break;
 #endif
+#ifdef CONFIG_TWO_SD_BOOT
+		case 5:
+			/* booting from Carrier SD*/
+			printf("booting from Carrier SD\n");
+			setenv("mmcdev", "2");
+			sprintf(buf, "/dev/mmcblk2p2 rootwait rw");
+			setenv("mmcroot",buf);
+                        break;
+#endif
 	}
 
 	/*record ddr bit, 32 or 64 bit*/
-- 
1.7.9.5

