From a40521f44f144e036f1948e3cd58d108b1516d0d Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 21 Jan 2016 17:02:39 +0800
Subject: [PATCH] twosd3


diff --git a/include/configs/mx6rom7421.h b/include/configs/mx6rom7421.h
index 219cc6c..1556a06 100755
--- a/include/configs/mx6rom7421.h
+++ b/include/configs/mx6rom7421.h
@@ -26,6 +26,10 @@
 /* support SATA boot */
 #define CONFIG_SATA_BOOT
 #define CONFIG_SATA_GEN2        0x0593e4c4
+
+/* support two sd card boot*/
+#define CONFIG_TWO_SD_BOOT
+
 #if defined(CONFIG_MX6QP)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6qp-rom7421-a1.dtb"
 #define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
@@ -124,7 +128,7 @@
 #define IOMUX_LCD_VDD_EN	MX6_PAD_NANDF_CLE__GPIO6_IO07
 #define LCD_BKLT_PWM 		IMX_GPIO_NR(1, 21)
 #define LCD_BKLT_EN 		IMX_GPIO_NR(6, 9)
-#define LCD_VDD_EN 		IMX_GPIO_NR(6, 7)	
+#define LCD_VDD_EN			IMX_GPIO_NR(6, 7)	
 #endif
 
 #define SPI1_CS0                IMX_GPIO_NR(3,19)
@@ -133,6 +137,7 @@
 #define IOMUX_SPI_MOSI          MX6_PAD_EIM_D18__ECSPI1_MOSI
 #define IOMUX_SPI_CS0           MX6_PAD_EIM_D19__ECSPI1_SS1
 
-#define USDHC2_CD_GPIO          IMX_GPIO_NR(2, 0)
-#define USDHC3_CD_GPIO          IMX_GPIO_NR(2, 1)
+#define USDHC2_CD_GPIO		IMX_GPIO_NR(2, 0)
+#define USDHC3_CD_GPIO		IMX_GPIO_NR(2, 1)
+#define USDHC3_PWREN_GPIO	IMX_GPIO_NR(2, 2)
 #endif                         /* __MX6QSABRESD_CONFIG_H */
-- 
1.7.9.5

