From a36f505d101ae2565953510436a66c9e90ab2fa4 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 17 Mar 2016 14:02:45 +0800
Subject: [PATCH] modify sd1


diff --git a/common/board_r.c b/common/board_r.c
old mode 100644
new mode 100755
index 8b5d15b..aa240cc
--- a/common/board_r.c
+++ b/common/board_r.c
@@ -904,7 +904,7 @@ int board_set_boot_device(void)
 			if(emmc_exist) 	setenv("mmcdev", "1");
 			break;
 #endif
-#ifdef CONFIG_TWO_SD_BOOT
+#if defined(USDHC2_CD_GPIO) && defined(USDHC3_CD_GPIO)
 		case 5:
 			/* booting from Carrier SD*/
 			printf("booting from Carrier SD\n");
-- 
1.7.9.5

