From f9a165c8d7778e668572a5dcce5d7642370fe508 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Sun, 10 Apr 2016 20:45:15 +0800
Subject: [PATCH] s4


diff --git a/include/spl.h b/include/spl.h
index 73edf13..a95a16d 100644
--- a/include/spl.h
+++ b/include/spl.h
@@ -71,7 +71,7 @@ void spl_net_load_image(const char *device);
 void spl_usb_load_image(void);
 
 /* SATA SPL functions */
-void spl_sata_load_image(void);
+int spl_sata_load_image(unsigned int dev);
 
 /* SPL FAT image functions */
 int spl_load_image_fat(block_dev_desc_t *block_dev, int partition, const char *filename);
-- 
1.7.9.5

