From cc584d944eb4b132e0a8cde552708c7362f9d39d Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Sun, 10 Apr 2016 20:43:14 +0800
Subject: [PATCH] s1


diff --git a/board/freescale/mx6advantech/mx6advantech.c b/board/freescale/mx6advantech/mx6advantech.c
index d62c475..4445710 100644
--- a/board/freescale/mx6advantech/mx6advantech.c
+++ b/board/freescale/mx6advantech/mx6advantech.c
@@ -943,6 +943,11 @@ int board_early_init_f(void)
 #ifdef CONFIG_ADVANTECH
 	enable_AXI_cache();
 #endif
+
+#ifdef CONFIG_CMD_SATA
+        setup_sata();
+#endif
+
 	return 0;
 }
 
@@ -984,10 +989,6 @@ int board_init(void)
 	setup_epdc();
 #endif
 
-#ifdef CONFIG_CMD_SATA
-	setup_sata();
-#endif
-
 	return 0;
 }
 #ifndef CONFIG_ADVANTECH
-- 
1.7.9.5

