From fdba2b0b239deb0cc06078f46c36c0b87f8918f4 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 14 Apr 2016 13:47:18 +0800
Subject: [PATCH] 7421bug


diff --git a/common/board_r.c b/common/board_r.c
index aa240cc..d88064b 100644
--- a/common/board_r.c
+++ b/common/board_r.c
@@ -895,20 +895,20 @@ int board_set_boot_device(void)
 		case 3:
 			/* booting from iNAND*/
 			printf("booting from iNAND\n");
-			setenv("mmcdev", "1");
+			setenv("mmcdev", MK_STR(CONFIG_EMMC_DEV_NUM));
 			break;
 #ifdef CONFIG_SPI_BOOT
 		case 4:
 			/* booting from SPI*/
 			printf("booting from SPI -> kernel boot form EMMC\n");
-			if(emmc_exist) 	setenv("mmcdev", "1");
+			if(emmc_exist) 	setenv("mmcdev", MK_STR(CONFIG_EMMC_DEV_NUM));
 			break;
 #endif
 #if defined(USDHC2_CD_GPIO) && defined(USDHC3_CD_GPIO)
 		case 5:
 			/* booting from Carrier SD*/
 			printf("booting from Carrier SD\n");
-			setenv("mmcdev", "2");
+			setenv("mmcdev", MK_STR(CONFIG_CARRIERSD_DEV_NUM));
 			sprintf(buf, "/dev/mmcblk2p2 rootwait rw");
 			setenv("mmcroot",buf);
                         break;
-- 
1.7.9.5

