From 55b5ca54ea6c7b49e5de9b1284018b3f0203b5a5 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 21 Apr 2016 17:31:13 +0800
Subject: [PATCH] 7421p2


diff --git a/common/board_r.c b/common/board_r.c
index d88064b..ee9faff 100644
--- a/common/board_r.c
+++ b/common/board_r.c
@@ -776,7 +776,7 @@ static int initr_check_fastboot(void)
 #ifdef CONFIG_ADVANTECH
 int check_emmc_exist(void)
 {
-        struct mmc *mmc = find_mmc_device(1);
+        struct mmc *mmc = find_mmc_device(CONFIG_EMMC_DEV_NUM);
         int result=0;
         if (mmc) {
                 result = mmc_init(mmc);
-- 
1.7.9.5

