From 65a07fe2f825a9edfabb3f56886fb8e3334b3f26 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Wed, 27 Apr 2016 14:23:08 +0800
Subject: [PATCH] spi3


diff --git a/include/configs/imx6_spl_advantech.h b/include/configs/imx6_spl_advantech.h
index 8792076..a4b65ad 100755
--- a/include/configs/imx6_spl_advantech.h
+++ b/include/configs/imx6_spl_advantech.h
@@ -54,6 +54,15 @@
 #define CONFIG_SPL_SATA_SUPPORT
 #endif
 
+/* SPI support */
+#ifdef CONFIG_SPI_BOOT
+#define CONFIG_SPL_SPI_SUPPORT
+#define CONFIG_SPL_SPI_FLASH_SUPPORT
+#define CONFIG_SPL_SPI_LOAD
+#define CONFIG_SYS_SPI_U_BOOT_OFFS      (512*3)
+/* Need to define CONFIG_SPL_SPI_BUS & CONFIG_SPL_SPI_CS in project H file */
+#endif
+
 #if defined(CONFIG_MX6SX)
 #define CONFIG_SPL_BSS_START_ADDR      0x88200000
 #define CONFIG_SPL_BSS_MAX_SIZE        0x100000        /* 1 MB */
diff --git a/include/configs/mx6rom3420.h b/include/configs/mx6rom3420.h
index 11b1055..7e2e6f2 100644
--- a/include/configs/mx6rom3420.h
+++ b/include/configs/mx6rom3420.h
@@ -14,6 +14,7 @@
 
 #ifdef CONFIG_SPL
 #define CONFIG_SATA_BOOT
+#define CONFIG_SPI_BOOT
 #define CONFIG_SPL_LIBCOMMON_SUPPORT
 #define CONFIG_SPL_LIBGENERIC_SUPPORT
 #define CONFIG_SPL_LIBDISK_SUPPORT
@@ -47,7 +48,6 @@
 #undef CONFIG_LDO_BYPASS_CHECK
 
 /* support SPI boot */
-#define CONFIG_SPI_BOOT
 #define CONFIG_SPL_SPI_BUS	0
 #define CONFIG_SPL_SPI_CS	0
 #define IOMUX_SPI5_SCLK         MX6_PAD_SD1_CLK__ECSPI5_SCLK
diff --git a/include/configs/mx6rom5420.h b/include/configs/mx6rom5420.h
index d62c2cd..bc20d72 100644
--- a/include/configs/mx6rom5420.h
+++ b/include/configs/mx6rom5420.h
@@ -14,6 +14,7 @@
 
 #ifdef CONFIG_SPL
 #define CONFIG_SATA_BOOT
+#define CONFIG_SPI_BOOT
 #define CONFIG_SPL_LIBCOMMON_SUPPORT
 #define CONFIG_SPL_LIBGENERIC_SUPPORT
 #define CONFIG_SPL_LIBDISK_SUPPORT
@@ -64,7 +65,6 @@
 #undef CONFIG_LDO_BYPASS_CHECK
 
 /* support SPI boot */
-#define CONFIG_SPI_BOOT
 #define CONFIG_SPL_SPI_BUS	4
 #define CONFIG_SPL_SPI_CS	0
 #define IOMUX_SPI5_SCLK		MX6_PAD_SD1_CLK__ECSPI5_SCLK
-- 
1.7.9.5

