From 7c81f1b95579264c5e5a3cd9db014a4c248526a1 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Wed, 27 Apr 2016 14:22:29 +0800
Subject: [PATCH] spi1


diff --git a/board/freescale/mx6advantech/mx6advantech.c b/board/freescale/mx6advantech/mx6advantech.c
index 6d6cd93..92a036b 100644
--- a/board/freescale/mx6advantech/mx6advantech.c
+++ b/board/freescale/mx6advantech/mx6advantech.c
@@ -193,7 +193,7 @@ static iomux_v3_cfg_t const usdhc4_pads[] = {
 	MX6_PAD_SD4_DAT7__SD4_DATA7 | MUX_PAD_CTRL(USDHC_PAD_CTRL),
 };
 
-#ifdef CONFIG_MXC_SPI
+#ifdef CONFIG_SYS_USE_SPINOR
 static iomux_v3_cfg_t const ecspi1_pads[] = {
         MX6_PAD_EIM_D16__ECSPI1_SCLK  | MUX_PAD_CTRL(SPI_PAD_CTRL),
         MX6_PAD_EIM_D17__ECSPI1_MISO  | MUX_PAD_CTRL(SPI_PAD_CTRL),
@@ -201,7 +201,7 @@ static iomux_v3_cfg_t const ecspi1_pads[] = {
         MX6_PAD_EIM_EB2__ECSPI1_SS0   | MUX_PAD_CTRL(NO_PAD_CTRL),
 };
 
-static void setup_spi(void)
+static void setup_spinor(void)
 {
 	imx_iomux_v3_setup_multiple_pads(ecspi1_pads, ARRAY_SIZE(ecspi1_pads));
 #ifdef SPI1_CS0
@@ -212,10 +212,6 @@ static void setup_spi(void)
 #endif
 }
 
-int board_spi_cs_gpio(unsigned bus, unsigned cs)
-{
-	return (bus == 0 && cs == 0) ? (IMX_GPIO_NR(4, 9)) : -1;
-}
 #endif
 
 static iomux_v3_cfg_t const rgb_pads[] = {
@@ -939,6 +935,9 @@ int board_early_init_f(void)
 #ifdef CONFIG_ADVANTECH
 	enable_AXI_cache();
 #endif
+#ifdef CONFIG_SYS_USE_SPINOR
+        setup_spinor();
+#endif
 
 #ifdef CONFIG_CMD_SATA
         setup_sata();
-- 
1.7.9.5

