From 6df7fc252f5f473fef9bca9e511340d4e8ffc3c2 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 28 Apr 2016 16:56:01 +0800
Subject: [PATCH 2/2] 7421-2


diff --git a/board/freescale/mx6advantech/Kconfig b/board/freescale/mx6advantech/Kconfig
index cca9dda..74cd34d 100644
--- a/board/freescale/mx6advantech/Kconfig
+++ b/board/freescale/mx6advantech/Kconfig
@@ -85,6 +85,12 @@ config SYS_CONFIG_NAME
 	default "mx6rom7421"
 
 endif
+if TARGET_MX6ROM7421A1_2G
+
+config SYS_CONFIG_NAME
+	default "mx6rom7421"
+
+endif
 if TARGET_MX6WISE3310A1_1G
 
 config SYS_CONFIG_NAME
-- 
1.7.9.5

