From 81eb2cda07c63e3c647b5ca3738876b4dad32662 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 28 Apr 2016 17:47:10 +0800
Subject: [PATCH] up


diff --git a/board/freescale/mx6advantech/mx6advantech.c b/board/freescale/mx6advantech/mx6advantech.c
index 92a036b..e366225 100644
--- a/board/freescale/mx6advantech/mx6advantech.c
+++ b/board/freescale/mx6advantech/mx6advantech.c
@@ -110,7 +110,18 @@ int enable_AXI_cache(void)
 
 int dram_init(void)
 {
+#ifdef CONFIG_ADVANTECH
+	/* Read memory size sent from Adv-Boot */
+	gd->ram_size = (*(unsigned int *)0x22400000);
+	if (gd->ram_size != (2u * 1024 * 1024 * 1024) &&
+		gd->ram_size != (1u * 1024 * 1024 * 1024) &&
+		gd->ram_size != (512 * 1024 * 1024))
+	{
+		gd->ram_size = PHYS_SDRAM_SIZE;
+	}
+#else
 	gd->ram_size = imx_ddr_size();
+#endif
 	return 0;
 }
 
-- 
1.7.9.5

