From 104e1189043eb9eaf950a174df4d291c489503cc Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 28 Apr 2016 17:51:41 +0800
Subject: [PATCH] up


diff --git a/include/configs/imx6_spl_advantech.h b/include/configs/imx6_spl_advantech.h
index a4b65ad..91ce390 100755
--- a/include/configs/imx6_spl_advantech.h
+++ b/include/configs/imx6_spl_advantech.h
@@ -34,6 +34,8 @@
 #define CONFIG_SPL_SERIAL_SUPPORT
 #define CONFIG_SPL_I2C_SUPPORT
 #define CONFIG_SPL_GPIO_SUPPORT
+#define CONFIG_SPL_BOARD_INIT
+
 
 /* NAND support */
 #if defined(CONFIG_SPL_NAND_SUPPORT)
diff --git a/include/configs/mx6rom3420.h b/include/configs/mx6rom3420.h
index 7e2e6f2..18228c1 100644
--- a/include/configs/mx6rom3420.h
+++ b/include/configs/mx6rom3420.h
@@ -28,19 +28,24 @@
 #define CONFIG_MMCROOT			"/dev/mmcblk0p2"  /* SDHC3 */
 /* support SATA boot */
 #define CONFIG_SATA_BOOT
-#define CONFIG_SATA_GEN2	0x059195fe	
+#define CONFIG_SATA_GEN2	0x059195fe
+
+#if defined(CONFIG_TARGET_MX6ROM3420A1_512M)
+#define PHYS_SDRAM_SIZE         (512u * 1024 * 1024)
+#elif defined(CONFIG_TARGET_MX6ROM3420A1_1G)
+#define PHYS_SDRAM_SIZE         (1u * 1024 * 1024 * 1024)
+#elif defined(CONFIG_TARGET_MX6ROM3420A1_2G)
+#define PHYS_SDRAM_SIZE         (2u * 1024 * 1024 * 1024)
+#endif
+	
 #if defined(CONFIG_MX6QP)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6qp-rom3420-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6Q)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6q-rom3420-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6DL)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-rom3420-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6SOLO)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-rom3420-a1.dtb"
-#define PHYS_SDRAM_SIZE		(512u * 1024 * 1024)
 #endif
 
 #include "mx6advantech_common.h"
diff --git a/include/configs/mx6rom5420.h b/include/configs/mx6rom5420.h
index bc20d72..64b4e08 100644
--- a/include/configs/mx6rom5420.h
+++ b/include/configs/mx6rom5420.h
@@ -30,33 +30,34 @@
 /* support SATA boot */
 #define CONFIG_SATA_BOOT
 #define CONFIG_SATA_GEN2        0x05919552
+
+#if defined(CONFIG_TARGET_MX6ROM5420A1_512M) || defined(CONFIG_TARGET_MX6ROM5420B1_512M)
+#define PHYS_SDRAM_SIZE		(512u * 1024 * 1024)
+#elif defined(CONFIG_TARGET_MX6ROM5420A1_1G) || defined(CONFIG_TARGET_MX6ROM5420B1_1G)
+#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
+#elif defined(CONFIG_TARGET_MX6ROM5420A1_2G) || defined(CONFIG_TARGET_MX6ROM5420B1_2G)
+#define PHYS_SDRAM_SIZE		(2u * 1024 * 1024 * 1024)
+#endif
+
 #if defined(CONFIG_TARGET_MX6ROM5420A1_1G) || defined(CONFIG_TARGET_MX6ROM5420A1_2G)
 #if defined(CONFIG_MX6QP)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6qp-rom5420-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6Q)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6q-rom5420-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6DL)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-rom5420-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6SOLO)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-rom5420-a1.dtb"
-#define PHYS_SDRAM_SIZE		(512u * 1024 * 1024)
 #endif
 #elif defined(CONFIG_TARGET_MX6ROM5420B1_1G) || defined(CONFIG_TARGET_MX6ROM5420B1_2G)
 #if defined(CONFIG_MX6QP)
 #define CONFIG_DEFAULT_FDT_FILE "imx6qp-rom5420-b1.dtb"
-#define PHYS_SDRAM_SIZE         (1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6Q)
 #define CONFIG_DEFAULT_FDT_FILE "imx6q-rom5420-b1.dtb"
-#define PHYS_SDRAM_SIZE         (1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6DL)
 #define CONFIG_DEFAULT_FDT_FILE "imx6dl-rom5420-b1.dtb"
-#define PHYS_SDRAM_SIZE         (1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6SOLO)
 #define CONFIG_DEFAULT_FDT_FILE "imx6dl-rom5420-b1.dtb"
-#define PHYS_SDRAM_SIZE         (512u * 1024 * 1024)
 #endif
 #endif
 
diff --git a/include/configs/mx6rom7420.h b/include/configs/mx6rom7420.h
index 938496a..d705e4e 100644
--- a/include/configs/mx6rom7420.h
+++ b/include/configs/mx6rom7420.h
@@ -29,18 +29,23 @@
 /* support SATA boot */
 #define CONFIG_SATA_BOOT
 #define CONFIG_SATA_GEN2        0x0593e4c4
+
+#if defined(CONFIG_TARGET_MX6ROM7420A1_512M)
+#define PHYS_SDRAM_SIZE         (512u * 1024 * 1024)
+#elif defined(CONFIG_TARGET_MX6ROM7420A1_1G)
+#define PHYS_SDRAM_SIZE         (1u * 1024 * 1024 * 1024)
+#elif defined(CONFIG_TARGET_MX6ROM7420A1_2G)
+#define PHYS_SDRAM_SIZE         (2u * 1024 * 1024 * 1024)
+#endif
+
 #if defined(CONFIG_MX6QP)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6qp-rom7420-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6Q)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6q-rom7420-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6DL)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-rom7420-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6SOLO)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-rom7420-a1.dtb"
-#define PHYS_SDRAM_SIZE		(512u * 1024 * 1024)
 #endif
 
 #include "mx6advantech_common.h"
diff --git a/include/configs/mx6rom7421.h b/include/configs/mx6rom7421.h
index 165c7c9..62dc8d4 100644
--- a/include/configs/mx6rom7421.h
+++ b/include/configs/mx6rom7421.h
@@ -30,18 +30,22 @@
 #define CONFIG_SATA_BOOT
 #define CONFIG_SATA_GEN2        0x0593e4c4
 
+#if defined(CONFIG_TARGET_MX6ROM7421A1_512M)
+#define PHYS_SDRAM_SIZE         (512u * 1024 * 1024)
+#elif defined(CONFIG_TARGET_MX6ROM7421A1_1G) 
+#define PHYS_SDRAM_SIZE         (1u * 1024 * 1024 * 1024)
+#elif defined(CONFIG_TARGET_MX6ROM7421A1_2G) 
+#define PHYS_SDRAM_SIZE         (2u * 1024 * 1024 * 1024)
+#endif
+
 #if defined(CONFIG_MX6QP)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6qp-rom7421-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6Q)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6q-rom7421-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6DL)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-rom7421-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6SOLO)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-rom7421-a1.dtb"
-#define PHYS_SDRAM_SIZE		(512u * 1024 * 1024)
 #endif
 
 #include "mx6advantech_common.h"
diff --git a/include/configs/mx6rsb4410.h b/include/configs/mx6rsb4410.h
index 3df785e..3d4619d 100644
--- a/include/configs/mx6rsb4410.h
+++ b/include/configs/mx6rsb4410.h
@@ -22,33 +22,34 @@
 #define CONFIG_MXC_UART_BASE	UART1_BASE
 #define CONFIG_CONSOLE_DEV		"ttymxc0"
 #define CONFIG_MMCROOT			"/dev/mmcblk0p2"  /* SDHC3 */
+
+#if defined(CONFIG_TARGET_MX6RSB4410A1_512M) || defined(CONFIG_TARGET_MX6RSB4410A2_512M)
+#define PHYS_SDRAM_SIZE         (512u * 1024 * 1024)
+#elif defined(CONFIG_TARGET_MX6RSB4410A1_1G) || defined(CONFIG_TARGET_MX6RSB4410A2_1G)
+#define PHYS_SDRAM_SIZE         (1u * 1024 * 1024 * 1024)
+#elif defined(CONFIG_TARGET_MX6RSB4410A1_2G) || defined(CONFIG_TARGET_MX6RSB4410A2_2G)
+#define PHYS_SDRAM_SIZE         (2u * 1024 * 1024 * 1024)
+#endif
+
 #if defined(CONFIG_TARGET_MX6RSB4410A1_1G)
 #if defined(CONFIG_MX6QP)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6qp-rsb4410-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6Q)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6q-rsb4410-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6DL)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-rsb4410-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6SOLO)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-rsb4410-a1.dtb"
-#define PHYS_SDRAM_SIZE		(512u * 1024 * 1024)
 #endif
 #elif defined(CONFIG_TARGET_MX6RSB4410A2_1G)
 #if defined(CONFIG_MX6QP)
 #define CONFIG_DEFAULT_FDT_FILE "imx6qp-rsb4410-a2.dtb"
-#define PHYS_SDRAM_SIZE         (1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6Q)
 #define CONFIG_DEFAULT_FDT_FILE "imx6q-rsb4410-a2.dtb"
-#define PHYS_SDRAM_SIZE         (1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6DL)
 #define CONFIG_DEFAULT_FDT_FILE "imx6dl-rsb4410-a2.dtb"
-#define PHYS_SDRAM_SIZE         (1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6SOLO)
 #define CONFIG_DEFAULT_FDT_FILE "imx6dl-rsb4410-a2.dtb"
-#define PHYS_SDRAM_SIZE         (512u * 1024 * 1024)
 #endif
 #endif
 
diff --git a/include/configs/mx6rsb4411.h b/include/configs/mx6rsb4411.h
index a7c59ba..44e1bd3 100755
--- a/include/configs/mx6rsb4411.h
+++ b/include/configs/mx6rsb4411.h
@@ -25,33 +25,34 @@
 #define CONFIG_MXC_UART_BASE	UART1_BASE
 #define CONFIG_CONSOLE_DEV		"ttymxc0"
 #define CONFIG_MMCROOT			"/dev/mmcblk0p2"  /* SDHC3 */
+
+#if defined(CONFIG_TARGET_MX6RSB4411A1_512M) || defined(CONFIG_TARGET_MX6RSB4411A2_512M)
+#define PHYS_SDRAM_SIZE         (512u * 1024 * 1024)
+#elif defined(CONFIG_TARGET_MX6RSB4411A1_1G) || defined(CONFIG_TARGET_MX6RSB4411A2_1G)
+#define PHYS_SDRAM_SIZE         (1u * 1024 * 1024 * 1024)
+#elif defined(CONFIG_TARGET_MX6RSB4411A1_2G) || defined(CONFIG_TARGET_MX6RSB4411A2_2G)
+#define PHYS_SDRAM_SIZE         (2u * 1024 * 1024 * 1024)
+#endif
+
 #if defined(CONFIG_TARGET_MX6RSB4411A1_1G)
 #if defined(CONFIG_MX6QP)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6qp-rsb4411-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6Q)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6q-rsb4411-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6DL)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-rsb4411-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6SOLO)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-rsb4411-a1.dtb"
-#define PHYS_SDRAM_SIZE		(512u * 1024 * 1024)
 #endif
 #elif defined(CONFIG_TARGET_MX6RSB4411A2_1G)
 #if defined(CONFIG_MX6QP)
 #define CONFIG_DEFAULT_FDT_FILE "imx6qp-rsb4411-a2.dtb"
-#define PHYS_SDRAM_SIZE         (1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6Q)
 #define CONFIG_DEFAULT_FDT_FILE "imx6q-rsb4411-a2.dtb"
-#define PHYS_SDRAM_SIZE         (1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6DL)
 #define CONFIG_DEFAULT_FDT_FILE "imx6dl-rsb4411-a2.dtb"
-#define PHYS_SDRAM_SIZE         (1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6SOLO)
 #define CONFIG_DEFAULT_FDT_FILE "imx6dl-rsb4411-a2.dtb"
-#define PHYS_SDRAM_SIZE         (512u * 1024 * 1024)
 #endif
 #endif
 
diff --git a/include/configs/mx6ubc220.h b/include/configs/mx6ubc220.h
index 21a4ba4..08f9920 100644
--- a/include/configs/mx6ubc220.h
+++ b/include/configs/mx6ubc220.h
@@ -22,18 +22,23 @@
 #define CONFIG_MXC_UART_BASE	UART1_BASE
 #define CONFIG_CONSOLE_DEV		"ttymxc0"
 #define CONFIG_MMCROOT			"/dev/mmcblk0p2"  /* SDHC3 */
+
+#if defined(CONFIG_TARGET_MX6UBC220A1_512M)
+#define PHYS_SDRAM_SIZE         (512u * 1024 * 1024)
+#elif defined(CONFIG_TARGET_MX6UBC220A1_1G)
+#define PHYS_SDRAM_SIZE         (1u * 1024 * 1024 * 1024)
+#elif defined(CONFIG_TARGET_MX6UBC220A1_2G)
+#define PHYS_SDRAM_SIZE         (2u * 1024 * 1024 * 1024)
+#endif
+
 #if defined(CONFIG_MX6QP)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6qp-ubc220-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6Q)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6q-ubc220-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6DL)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-ubc220-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6SOLO)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-ubc220-a1.dtb"
-#define PHYS_SDRAM_SIZE		(512u * 1024 * 1024)
 #endif
 
 #include "mx6advantech_common.h"
diff --git a/include/configs/mx6ubcds31.h b/include/configs/mx6ubcds31.h
index 1bac0ba..e5602a2 100644
--- a/include/configs/mx6ubcds31.h
+++ b/include/configs/mx6ubcds31.h
@@ -22,18 +22,23 @@
 #define CONFIG_MXC_UART_BASE	UART1_BASE
 #define CONFIG_CONSOLE_DEV		"ttymxc0"
 #define CONFIG_MMCROOT			"/dev/mmcblk0p2"  /* SDHC3 */
+
+#if defined(CONFIG_TARGET_MX6UBCDS31A1_512M)
+#define PHYS_SDRAM_SIZE         (512u * 1024 * 1024)
+#elif defined(CONFIG_TARGET_MX6UBCDS31A1_1G)
+#define PHYS_SDRAM_SIZE         (1u * 1024 * 1024 * 1024)
+#elif defined(CONFIG_TARGET_MX6UBCDS31A1_2G)
+#define PHYS_SDRAM_SIZE         (2u * 1024 * 1024 * 1024)
+#endif
+
 #if defined(CONFIG_MX6QP)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6qp-ubcds31-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6Q)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6q-ubcds31-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6DL)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-ubcds31-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6SOLO)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-ubcds31-a1.dtb"
-#define PHYS_SDRAM_SIZE		(512u * 1024 * 1024)
 #endif
 
 #include "mx6advantech_common.h"
diff --git a/include/configs/mx6wise3310.h b/include/configs/mx6wise3310.h
index e6cb8ee..b278f17 100644
--- a/include/configs/mx6wise3310.h
+++ b/include/configs/mx6wise3310.h
@@ -22,18 +22,23 @@
 #define CONFIG_MXC_UART_BASE	UART1_BASE
 #define CONFIG_CONSOLE_DEV		"ttymxc0"
 #define CONFIG_MMCROOT			"/dev/mmcblk0p2"  /* SDHC3 */
+
+#if defined(CONFIG_TARGET_MX6WISE3310A1_512M)
+#define PHYS_SDRAM_SIZE         (512u * 1024 * 1024)
+#elif defined(CONFIG_TARGET_MX6WISE3310A1_1G)
+#define PHYS_SDRAM_SIZE         (1u * 1024 * 1024 * 1024)
+#elif defined(CONFIG_TARGET_MX6WISE3310A1_2G)
+#define PHYS_SDRAM_SIZE         (2u * 1024 * 1024 * 1024)
+#endif
+
 #if defined(CONFIG_MX6QP)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6qp-wise3310-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6Q)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6q-wise3310-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6DL)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-wise3310-a1.dtb"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
 #elif defined(CONFIG_MX6SOLO)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-wise3310-a1.dtb"
-#define PHYS_SDRAM_SIZE		(512u * 1024 * 1024)
 #endif
 
 #include "mx6advantech_common.h"
-- 
1.7.9.5

