From a542c910d6813a287dc2ac9b524b1c8ddb6b107f Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 28 Apr 2016 17:26:38 +0800
Subject: [PATCH 1/4] 1


diff --git a/arch/arm/imx-common/spl_advantech.c b/arch/arm/imx-common/spl_advantech.c
index 866d98b..50e718d 100644
--- a/arch/arm/imx-common/spl_advantech.c
+++ b/arch/arm/imx-common/spl_advantech.c
@@ -141,7 +141,7 @@ void spl_board_init(void)
 	}
 
 	/* forword memory size to uboot */
-	//*(unsigned int *)0x22400000 = PHYS_SDRAM_1_SIZE;
+	*(unsigned int *)0x22400000 = PHYS_SDRAM_SIZE;
 
 	/* record ddr bit, 32 or 64 bit */
 #ifdef CONFIG_DDR_32BIT 
-- 
1.7.9.5

