From 087f3a5a75108d7100907c4aeff9f7e579f1b8d7 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Fri, 29 Apr 2016 11:19:23 +0800
Subject: [PATCH] Modify sata gen2 register value


diff --git a/include/configs/mx6rom3420.h b/include/configs/mx6rom3420.h
index 18228c1..74d1847 100644
--- a/include/configs/mx6rom3420.h
+++ b/include/configs/mx6rom3420.h
@@ -28,7 +28,7 @@
 #define CONFIG_MMCROOT			"/dev/mmcblk0p2"  /* SDHC3 */
 /* support SATA boot */
 #define CONFIG_SATA_BOOT
-#define CONFIG_SATA_GEN2	0x059195fe
+#define CONFIG_SATA_GEN2	0x05918732
 
 #if defined(CONFIG_TARGET_MX6ROM3420A1_512M)
 #define PHYS_SDRAM_SIZE         (512u * 1024 * 1024)
diff --git a/include/configs/mx6rom7421.h b/include/configs/mx6rom7421.h
index 62dc8d4..6ad21db 100644
--- a/include/configs/mx6rom7421.h
+++ b/include/configs/mx6rom7421.h
@@ -28,7 +28,7 @@
 
 /* support SATA boot */
 #define CONFIG_SATA_BOOT
-#define CONFIG_SATA_GEN2        0x0593e4c4
+#define CONFIG_SATA_GEN2        0x059194f6
 
 #if defined(CONFIG_TARGET_MX6ROM7421A1_512M)
 #define PHYS_SDRAM_SIZE         (512u * 1024 * 1024)
-- 
1.7.9.5

