From fdfb3e79bd7f93fade7dfa267b69dd0015c7b58b Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 26 May 2016 14:26:22 +0800
Subject: [PATCH 2/4] solo


diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index ab3baea..3051896 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -568,6 +568,11 @@ config TARGET_MX6ROM5420A1_2G
 	select CPU_V7
 	select SUPPORT_SPL
 
+config TARGET_MX6ROM5420B1_512M
+	bool "Support mx6rom5420b1 512M"
+	select CPU_V7
+	select SUPPORT_SPL
+
 config TARGET_MX6ROM5420B1_1G
 	bool "Support mx6rom5420b1 1G"
 	select CPU_V7
-- 
1.7.9.5

