From 91a5a1c3fc93842bf35dcb78c136513d501944ac Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 26 May 2016 14:26:37 +0800
Subject: [PATCH 3/4] solo


diff --git a/include/configs/imx6_spl_advantech.h b/include/configs/imx6_spl_advantech.h
index 91ce390..5623cf1 100755
--- a/include/configs/imx6_spl_advantech.h
+++ b/include/configs/imx6_spl_advantech.h
@@ -58,12 +58,14 @@
 
 /* SPI support */
 #ifdef CONFIG_SPI_BOOT
+#ifdef CONFIG_MX6Q
 #define CONFIG_SPL_SPI_SUPPORT
 #define CONFIG_SPL_SPI_FLASH_SUPPORT
 #define CONFIG_SPL_SPI_LOAD
 #define CONFIG_SYS_SPI_U_BOOT_OFFS      (512*3)
 /* Need to define CONFIG_SPL_SPI_BUS & CONFIG_SPL_SPI_CS in project H file */
 #endif
+#endif
 
 #if defined(CONFIG_MX6SX)
 #define CONFIG_SPL_BSS_START_ADDR      0x88200000
diff --git a/include/configs/mx6rom5420.h b/include/configs/mx6rom5420.h
index 64b4e08..cfd0d60 100644
--- a/include/configs/mx6rom5420.h
+++ b/include/configs/mx6rom5420.h
@@ -39,7 +39,7 @@
 #define PHYS_SDRAM_SIZE		(2u * 1024 * 1024 * 1024)
 #endif
 
-#if defined(CONFIG_TARGET_MX6ROM5420A1_1G) || defined(CONFIG_TARGET_MX6ROM5420A1_2G)
+#if defined(CONFIG_TARGET_MX6ROM5420A1_1G) || defined(CONFIG_TARGET_MX6ROM5420A1_2G) || defined(CONFIG_TARGET_MX6ROM5420A1_512M)
 #if defined(CONFIG_MX6QP)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6qp-rom5420-a1.dtb"
 #elif defined(CONFIG_MX6Q)
@@ -49,7 +49,7 @@
 #elif defined(CONFIG_MX6SOLO)
 #define CONFIG_DEFAULT_FDT_FILE	"imx6dl-rom5420-a1.dtb"
 #endif
-#elif defined(CONFIG_TARGET_MX6ROM5420B1_1G) || defined(CONFIG_TARGET_MX6ROM5420B1_2G)
+#elif defined(CONFIG_TARGET_MX6ROM5420B1_1G) || defined(CONFIG_TARGET_MX6ROM5420B1_2G) || defined(CONFIG_TARGET_MX6ROM5420B1_512M)
 #if defined(CONFIG_MX6QP)
 #define CONFIG_DEFAULT_FDT_FILE "imx6qp-rom5420-b1.dtb"
 #elif defined(CONFIG_MX6Q)
@@ -66,12 +66,14 @@
 #undef CONFIG_LDO_BYPASS_CHECK
 
 /* support SPI boot */
+#ifdef CONFIG_MX6Q
 #define CONFIG_SPL_SPI_BUS	4
 #define CONFIG_SPL_SPI_CS	0
 #define IOMUX_SPI5_SCLK		MX6_PAD_SD1_CLK__ECSPI5_SCLK
 #define IOMUX_SPI5_MISO		MX6_PAD_SD1_DAT0__ECSPI5_MISO
 #define IOMUX_SPI5_MOSI		MX6_PAD_SD1_CMD__ECSPI5_MOSI
 #define IOMUX_SPI5_CS0		MX6_PAD_SD1_DAT1__ECSPI5_SS0
+#endif
 #ifdef CONFIG_SPL_BUILD
 #define CONFIG_BOOT_SELECT
 #endif
-- 
1.7.9.5

