From 8fb4fecd4321e46fea1af91854da7b6e91278d10 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 30 Jun 2016 16:35:33 +0800
Subject: [PATCH] splash


diff --git a/include/configs/mx6advantech_common.h b/include/configs/mx6advantech_common.h
index 7784e8e..242b5c5 100644
--- a/include/configs/mx6advantech_common.h
+++ b/include/configs/mx6advantech_common.h
@@ -479,27 +479,6 @@
 #define CONFIG_SYS_I2C_MXC
 #define CONFIG_SYS_I2C_SPEED		  100000
 
-/* Framebuffer */
-#define CONFIG_VIDEO
-#define CONFIG_VIDEO_IPUV3
-#define CONFIG_CFB_CONSOLE
-#define CONFIG_VGA_AS_SINGLE_DEVICE
-#define CONFIG_SYS_CONSOLE_IS_IN_ENV
-#define CONFIG_SYS_CONSOLE_OVERWRITE_ROUTINE
-#define CONFIG_VIDEO_BMP_RLE8
-#define CONFIG_SPLASH_SCREEN
-#define CONFIG_SPLASH_SCREEN_ALIGN
-#define CONFIG_BMP_16BPP
-#define CONFIG_VIDEO_LOGO
-#define CONFIG_VIDEO_BMP_LOGO
-#ifdef CONFIG_MX6DL                                                             
-#define CONFIG_IPUV3_CLK 198000000
-#else
-#define CONFIG_IPUV3_CLK 264000000
-#endif
-#define CONFIG_IMX_HDMI
-#define CONFIG_IMX_VIDEO_SKIP
-
 #if defined(CONFIG_ANDROID_SUPPORT)
 #include "mx6sabreandroid_common.h"
 #endif
-- 
1.7.9.5

