From b1c379cca2f6cd4b883541f8c472d8cd600b57b2 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Tue, 23 Aug 2016 14:54:49 +0800
Subject: [PATCH] Fixed build u-boot failed issue for WISE-3310


diff --git a/include/configs/mx6wise3310.h b/include/configs/mx6wise3310.h
index b278f17..edabd11 100644
--- a/include/configs/mx6wise3310.h
+++ b/include/configs/mx6wise3310.h
@@ -112,9 +112,6 @@
 #endif /* CONFIG_SPLASH_SCREEN && CONFIG_MXC_EPDC */
 #endif
 
-
-#endif
-
 #define SPI1_CS0                IMX_GPIO_NR(2,30)
 #define IOMUX_SPI_SCLK          MX6_PAD_EIM_D16__ECSPI1_SCLK
 #define IOMUX_SPI_MISO          MX6_PAD_EIM_D17__ECSPI1_MISO
-- 
1.7.9.5

