From af5f83548c802ca89f42e2a9c15a32718a6d98f4 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 1 Sep 2016 13:56:52 +0800
Subject: [PATCH] add


diff --git a/usb_mount b/usb_mount
new file mode 100644
index 0000000..7777708
--- /dev/null
+++ b/usb_mount
@@ -0,0 +1,26 @@
+#!/bin/bash
+{
+V_YOCTO=`cat /etc/issue |grep Distro`
+
+if [ -n "$V_YOCTO" ] ;then # Yocto
+        if [ -d "/media/sda1" ] ;then
+                USB_DEV=/media/sda1
+        fi
+        if [ -d "/run/media/sda1" ] ;then
+                USB_DEV=/run/media/sda1
+        fi
+else
+        USB_DEV=/mnt/disk
+        if [ ! -d $USB_DEV ]; then
+            mkdir $USB_DEV
+        fi
+        /bin/mount /dev/udisk$1 $USB_DEV
+fi
+
+sleep 3
+
+# For Burn In Test
+if [ -e $USB_DEV/burnin.sh ]; then
+    $USB_DEV/burnin.sh
+fi
+} &
-- 
1.7.9.5

