diff --git a/core/arch/arm/plat-imx/conf.mk b/core/arch/arm/plat-imx/conf.mk
index fc934d3..915fda6 100644
--- a/core/arch/arm/plat-imx/conf.mk
+++ b/core/arch/arm/plat-imx/conf.mk
@@ -14,7 +14,7 @@ mx6ull-flavorlist = mx6ullevk
 mx7d-flavorlist = mx7dsabresd
 mx7s-flavorlist = mx7swarp7
 mx7ulp-flavorlist = mx7ulpevk
-mx8m-flavorlist = mx8mqevk
+mx8m-flavorlist = mx8mqevk mx8mqrom5720a1
 mx8mm-flavorlist = mx8mmevk
 mx8qm-flavorlist = mx8qmmek mx8qmlpddr4arm2
 mx8qx-flavorlist = mx8qxpmek mx8qxplpddr4arm2
@@ -423,6 +423,11 @@ CFG_DDR_SIZE ?= 0xC0000000
 CFG_UART_BASE ?= UART1_BASE
 endif
 
+ifneq (,$(filter $(PLATFORM_FLAVOR),mx8mqrom5720a1))
+CFG_DDR_SIZE ?= 0x80000000
+CFG_UART_BASE ?= UART1_BASE
+endif
+
 ifneq (,$(filter $(PLATFORM_FLAVOR),mx8mmevk))
 CFG_DDR_SIZE ?= 0x80000000
 CFG_UART_BASE ?= UART2_BASE
diff --git a/core/arch/arm/plat-imx/tzasc.c b/core/arch/arm/plat-imx/tzasc.c
index b2b7653..ed1aafd 100644
--- a/core/arch/arm/plat-imx/tzasc.c
+++ b/core/arch/arm/plat-imx/tzasc.c
@@ -310,7 +310,7 @@ static int board_imx_tzasc_configure(vaddr_t addr)
 
 	return 0;
 }
-#elif defined(PLATFORM_FLAVOR_mx8mqevk)
+#elif defined(PLATFORM_FLAVOR_mx8mqevk) || defined(PLATFORM_FLAVOR_mx8mqrom5720a1)
 static int board_imx_tzasc_configure(vaddr_t addr)
 {
 	tzc_init(addr);
