diff --git a/config.mk b/config.mk
index c5d913b..4b695b2 100644
--- a/config.mk
+++ b/config.mk
@@ -52,7 +52,7 @@ WITH_PERSISTENCE:=yes
 # Comment out to remove memory tracking support from the broker. If disabled,
 # mosquitto won't track heap memory usage nor export '$SYS/broker/heap/current
 # size', but will use slightly less memory and CPU time.
-WITH_MEMORY_TRACKING:=yes
+#WITH_MEMORY_TRACKING:=yes
 
 # Compile with database upgrading support? If disabled, mosquitto won't
 # automatically upgrade old database versions.
@@ -205,6 +205,6 @@ endif
 
 
 INSTALL?=install
-prefix=/usr/local
+prefix=/usr
 mandir=${prefix}/share/man
 localedir=${prefix}/share/locale
