diff --git a/client/Makefile b/client/Makefile
index 2df2c02..3d7fee7 100644
--- a/client/Makefile
+++ b/client/Makefile
@@ -21,8 +21,8 @@ sub_client.o : sub_client.c ../lib/libmosquitto.so.${SOVERSION}
 
 install : all
 	$(INSTALL) -d ${DESTDIR}$(prefix)/bin
-	$(INSTALL) -s --strip-program=$(STRIP) mosquitto_pub ${DESTDIR}${prefix}/bin/mosquitto_pub
-	$(INSTALL) -s --strip-program=$(STRIP) mosquitto_sub ${DESTDIR}${prefix}/bin/mosquitto_sub
+	$(INSTALL) mosquitto_pub ${DESTDIR}${prefix}/bin/mosquitto_pub
+	$(INSTALL) mosquitto_sub ${DESTDIR}${prefix}/bin/mosquitto_sub
 
 uninstall :
 	-rm -f ${DESTDIR}${prefix}/bin/mosquitto_pub
diff --git a/config.mk b/config.mk
index c47acaa..63d5072 100644
--- a/config.mk
+++ b/config.mk
@@ -52,7 +52,7 @@ WITH_PERSISTENCE:=yes
 # Comment out to remove memory tracking support from the broker. If disabled,
 # mosquitto won't track heap memory usage nor export '$SYS/broker/heap/current
 # size', but will use slightly less memory and CPU time.
-WITH_MEMORY_TRACKING:=yes
+#WITH_MEMORY_TRACKING:=yes
 
 # Compile with database upgrading support? If disabled, mosquitto won't
 # automatically upgrade old database versions.
@@ -65,10 +65,10 @@ WITH_SYS_TREE:=yes
 
 # Build with Python module. Comment out if Python is not installed, or required
 # Python modules are not available.
-WITH_PYTHON:=yes
+WITH_PYTHON:=no
 
 # Build with SRV lookup support.
-WITH_SRV:=yes
+WITH_SRV:=no
 
 # =============================================================================
 # End of user configuration
@@ -213,7 +213,7 @@ endif
 
 
 INSTALL?=install
-prefix=/usr/local
+prefix=/usr
 mandir=${prefix}/share/man
 localedir=${prefix}/share/locale
 STRIP?=strip
diff --git a/lib/Makefile b/lib/Makefile
index f3f048b..3281919 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -25,7 +25,7 @@ all : libmosquitto.so.${SOVERSION} libmosquitto.a
 
 install : all
 	$(INSTALL) -d ${DESTDIR}$(prefix)/lib${LIB_SUFFIX}/
-	$(INSTALL) -s --strip-program=$(STRIP) libmosquitto.so.${SOVERSION} ${DESTDIR}${prefix}/lib${LIB_SUFFIX}/libmosquitto.so.${SOVERSION}
+	$(INSTALL) libmosquitto.so.${SOVERSION} ${DESTDIR}${prefix}/lib${LIB_SUFFIX}/libmosquitto.so.${SOVERSION}
 	ln -sf libmosquitto.so.${SOVERSION} ${DESTDIR}${prefix}/lib${LIB_SUFFIX}/libmosquitto.so
 	$(INSTALL) -d ${DESTDIR}${prefix}/include/
 	$(INSTALL) mosquitto.h ${DESTDIR}${prefix}/include/mosquitto.h
diff --git a/lib/cpp/Makefile b/lib/cpp/Makefile
index dc609bd..641b1cd 100644
--- a/lib/cpp/Makefile
+++ b/lib/cpp/Makefile
@@ -10,7 +10,7 @@ all : libmosquittopp.so.${SOVERSION}
 
 install : all
 	$(INSTALL) -d ${DESTDIR}$(prefix)/lib${LIB_SUFFIX}/
-	$(INSTALL) -s --strip-program=$(STRIP) libmosquittopp.so.${SOVERSION} ${DESTDIR}${prefix}/lib${LIB_SUFFIX}/libmosquittopp.so.${SOVERSION}
+	$(INSTALL) libmosquittopp.so.${SOVERSION} ${DESTDIR}${prefix}/lib${LIB_SUFFIX}/libmosquittopp.so.${SOVERSION}
 	ln -sf libmosquittopp.so.${SOVERSION} ${DESTDIR}${prefix}/lib${LIB_SUFFIX}/libmosquittopp.so
 	$(INSTALL) -d ${DESTDIR}${prefix}/include/
 	$(INSTALL) mosquittopp.h ${DESTDIR}${prefix}/include/mosquittopp.h
diff --git a/src/Makefile b/src/Makefile
index e3a562d..b540af8 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -100,10 +100,10 @@ mosquitto_passwd.o : mosquitto_passwd.c
 
 install : all
 	$(INSTALL) -d ${DESTDIR}$(prefix)/sbin
-	$(INSTALL) -s --strip-program=$(STRIP) mosquitto ${DESTDIR}${prefix}/sbin/mosquitto
+	$(INSTALL) mosquitto ${DESTDIR}${prefix}/sbin/mosquitto
 	$(INSTALL) mosquitto_plugin.h ${DESTDIR}${prefix}/include/mosquitto_plugin.h
 ifeq ($(WITH_TLS),yes)
-	$(INSTALL) -s --strip-program=$(STRIP) mosquitto_passwd ${DESTDIR}${prefix}/bin/mosquitto_passwd
+	$(INSTALL) mosquitto_passwd ${DESTDIR}${prefix}/bin/mosquitto_passwd
 endif
 
 uninstall :
