Index: udev-extraconf/mount.sh
===================================================================
--- udev-extraconf/mount.sh	(revision 1421)
+++ udev-extraconf/mount.sh	(working copy)
@@ -25,7 +25,11 @@
 	if [ "x`readlink $MOUNT`" = "x/bin/mount.util-linux" ] ;
 	then
 		MOUNT="$MOUNT -o silent"
+		if [ "$name"=="sd*[0-9]" ]; then #[Adv] for burn-in
+			/tools/usb_mount
+		fi
 	fi
+
 	
 	# If filesystem type is vfat, change the ownership group to 'disk', and
 	# grant it with  w/r/x permissions.
