---
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure_1.ac
@@ -20,7 +20,7 @@
 AC_PATH_PROG(mknod, mknod, /bin/mknod)
 AC_PATH_PROG(mkfifo, mkfifo, /usr/bin/mkfifo)
 AC_PATH_PROG(depmod, depmod, /sbin/depmod, $PATH:/sbin)
-AC_PATH_PROG(LIBUSB_CONFIG, libusb-config)
+LIBUSB_CONFIG="pkg-config libusb"
 AC_PROG_LN_S
 AC_PROG_LIBTOOL
 AM_PATH_PYTHON(,, [:])
@@ -1789,6 +1789,7 @@
   tools/Makefile
   doc/Makefile
   doc/man/Makefile
+  conf/Makefile
 ])
 AC_OUTPUT

---
diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -6,10 +6,10 @@
 ## we need version 1.5 of automake for DIST_SUBDIRS to work and dist-bzip2.
 AUTOMAKE_OPTIONS = 1.5 check-news dist-bzip2
 
-EXTRA_DIST =  setup.sh setup-driver.sh setup.data data2setup.sh ANNOUNCE TODO
+EXTRA_DIST =  setup.sh setup-driver.sh setup.data data2setup.sh ANNOUNCE TODO conf
 
 ## daemons/ and tools/ should appear before doc/ for help2man to work.
-SUBDIRS = drivers daemons tools doc
+SUBDIRS = drivers daemons tools doc conf
 
 ## From automake documentation:
 ## Note that EXTRA_DIST can only handle files in the current
 
 ---
diff --git a/conf/Makefile.am b/conf/Makefile.am
--- a/conf/Makefile.am
+++ b/conf/Makefile.am
@@ -0,0 +1,2 @@
+lircdir = /etc
+lirc_DATA = lircd.conf lircrc

---
diff --git a/daemons/Makefile.am b/daemons/Makefile.am
--- a/daemons/Makefile.am 
+++ b/daemons/Makefile.am
@@ -70,6 +70,7 @@
                input_map.c input_map.h \
                transmit.c transmit.h
 lircd_LDADD = @daemon@ libhw_module.a @hw_module_libs@
+irrecord_DEPENDENCIES = libhw_module.a @receive@

 lircmd_SOURCES = lircmd.c
 lircmd_LDADD = @daemon@
@@ -83,7 +84,7 @@
                transmit.c transmit.h

 irrecord_LDADD = libhw_module.a @hw_module_libs@ @receive@
-irrecord_DEPENDENCIES = @receive@
+irrecord_DEPENDENCIES = libhw_module.a @receive@
