---
diff --git a/conf/lircrc b/conf/lircrc
--- a/conf/lircrc	(revision 192)
+++ b/conf/lircrc	(working copy)
@@ -0,0 +1,104 @@
+begin
+	remote = IR_CONTROL 
+	button = KEY_POWER
+	prog = irexec
+	config = echo "KEY_POWER"
+end
+
+begin
+	remote = IR_CONTROL 
+	button = KEY_2
+	prog = irexec
+	config = echo "KEY_2"
+end
+
+begin
+	remote = IR_CONTROL 
+	button = KEY_3
+	prog = irexec
+	config = echo "KEY_3"
+end
+
+begin
+	remote = IR_CONTROL 
+	button = KEY_4
+	prog = irexec
+	config = echo "KEY_4"
+end
+
+begin
+	remote = IR_CONTROL 
+	button = KEY_5
+	prog = irexec
+	config = echo "KEY_5"
+end
+
+begin
+	remote = IR_CONTROL 
+	button = KEY_6
+	prog = irexec
+	config = echo "KEY_6"
+end
+
+begin
+	remote = IR_CONTROL 
+	button = KEY_7
+	prog = irexec
+	config = echo "KEY_7"
+end
+
+begin
+	remote = IR_CONTROL 
+	button = KEY_8
+	prog = irexec
+	config = echo "KEY_8"
+end
+
+begin
+	remote = IR_CONTROL 
+	button = KEY_9
+	prog = irexec
+	config = echo "KEY_9"
+end
+
+begin
+	remote = IR_CONTROL 
+	button = KEY_A
+	prog = irexec
+	config = echo "KEY_A"
+end
+
+begin
+	remote = IR_CONTROL 
+	button = KEY_B
+	prog = irexec
+	config = echo "KEY_B"
+end
+
+begin
+	remote = IR_CONTROL 
+	button = KEY_C
+	prog = irexec
+	config = echo "KEY_C"
+end
+
+begin
+	remote = IR_CONTROL 
+	button = KEY_D
+	prog = irexec
+	config = echo "KEY_D"
+end
+
+begin
+	remote = IR_CONTROL 
+	button = KEY_E
+	prog = irexec
+	config = echo "KEY_E"
+end
+
+begin
+	remote = IR_CONTROL 
+	button = KEY_F
+	prog = irexec
+	config = echo "KEY_F"
+end
