---
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure_1.ac
@@ -360,6 +360,7 @@
   lib/Makefile
   doc/Makefile
   systemd/Makefile
+  conf/Makefile
 ])
 
 AC_OUTPUT

---
diff --git a/conf/Makefile.am b/conf/Makefile.am
--- a/conf/Makefile.am
+++ b/conf/Makefile.am
@@ -0,0 +1,2 @@
+lircdir = /etc
+lirc_DATA = lircd.conf lircrc

---
diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -4,7 +4,7 @@
 
 AUTOMAKE_OPTIONS        = 1.5 check-news dist-bzip2 -Wno-portability
 
-EXTRA_DIST              = README autogen.sh CONTRIBUTE.md ChangeLog
+EXTRA_DIST              = README autogen.sh CONTRIBUTE.md ChangeLog conf
 
 DISTCLEANFILES          = configure.sh  ChangeLog paths.h
 CLEANFILES              = *~ remotes/*~ contrib/*~
@@ -21,7 +21,7 @@
     SYSTEMD_DIR         = systemd
 endif
 
-SUBDIRS                 = lib daemons tools doc $(SYSTEMD_DIR) plugins
+SUBDIRS                 = lib daemons tools doc $(SYSTEMD_DIR) plugins conf
 

