Index: init-ifupdown-1.0/interfaces
===================================================================
--- init-ifupdown-1.0/interfaces	(revision 1182)
+++ init-ifupdown-1.0/interfaces	(working copy)
@@ -14,8 +14,11 @@
 iface atml0 inet dhcp

 # Wired or wireless interfaces
-auto eth0
+auto eth0 eth0:0
 iface eth0 inet dhcp
+iface eth0:0 inet static
+       address 192.168.0.1
+       netmask 255.255.255.0
 iface eth1 inet dhcp

 # Ethernet/RNDIS gadget (g_ether)
 
