diff --git a/plat/imx/imx8mq/imx8mq_bl31_setup.c b/plat/imx/imx8mq/imx8mq_bl31_setup.c
index 155a11a..a0123cb 100644
--- a/plat/imx/imx8mq/imx8mq_bl31_setup.c
+++ b/plat/imx/imx8mq/imx8mq_bl31_setup.c
@@ -307,11 +307,11 @@ void bl31_early_platform_setup2(u_register_t arg0, u_register_t arg1,
 	bl32_image_ep_info.args.arg1 = BL32_BASE;
 #else
 	/* Pass TEE base and size to uboot */
-	bl33_image_ep_info.args.arg1 = 0xFE000000;
+	bl33_image_ep_info.args.arg1 = 0xBE000000;
 #endif
 	/* TEE size + RDC reserved memory = 0x2000000 + 0x2000000 + 0x30000000 */
 #ifdef DECRYPTED_BUFFER_START
-	bl33_image_ep_info.args.arg2 = 0x100000000 - DECRYPTED_BUFFER_START;
+	bl33_image_ep_info.args.arg2 = 0xC0000000 - DECRYPTED_BUFFER_START;
 #else
 	bl33_image_ep_info.args.arg2 = 0x2000000;
 #endif
diff --git a/plat/imx/imx8mq/include/platform_def.h b/plat/imx/imx8mq/include/platform_def.h
index a25636e..551b5bc 100644
--- a/plat/imx/imx8mq/include/platform_def.h
+++ b/plat/imx/imx8mq/include/platform_def.h
@@ -31,7 +31,7 @@
 
 #define BL31_BASE			0x910000
 #define BL31_LIMIT			0x920000
-#define BL32_BASE			0xfe000000
+#define BL32_BASE			0xbe000000
 #ifdef SPD_trusty
 #define BL32_SIZE			0x02000000
 #define BL32_LIMIT			0x100000000
