From 171a33f1492c0baba7b8f0005a8ebb704498b29e Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 19 May 2016 15:18:32 +0800
Subject: [PATCH] add all risc projects to build dts file


diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index d4f292a..1f106cd 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -397,6 +397,16 @@ dtb-$(CONFIG_SOC_IMX7D) += \
 	imx7d-sdb-reva-qspi.dtb \
 	imx7d-sdb-reva-touch.dtb \
 	imx7d-sdb-reva-wm8960.dtb
+dtb-$(CONFIG_ARCH_ADVANTECH) += \
+	imx6dl-ubc220-a1.dtb \
+	imx6q-ubcds31-a1.dtb \
+	imx6q-rom3420-a1.dtb \
+	imx6q-rsb4410-a1.dtb \
+	imx6q-rsb4410-a2.dtb \
+	imx6q-rom5420-a1.dtb \
+	imx6q-rom5420-b1.dtb \
+	imx6q-rom7420-a1.dtb \
+	imx6q-wise3310-a1.dtb
 dtb-$(CONFIG_SOC_LS1021A) += \
 	ls1021a-qds.dtb \
 	ls1021a-twr.dtb
