From 033b20f3adcd3f45c46eb7136c4c302247746a4b Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Fri, 20 May 2016 10:35:59 +0800
Subject: [PATCH] Add sleep key function inculde suspend/wakeup


diff --git a/kernel/power/process.c b/kernel/power/process.c
old mode 100644
new mode 100755
index 564f786..7084dc7
--- a/kernel/power/process.c
+++ b/kernel/power/process.c
@@ -23,6 +23,9 @@
  * Timeout for stopping processes
  */
 unsigned int __read_mostly freeze_timeout_msecs = 20 * MSEC_PER_SEC;
+#ifdef CONFIG_ARCH_ADVANTECH
+extern char suspend_key_flag;
+#endif
 
 static int try_to_freeze_tasks(bool user_only)
 {
@@ -212,6 +215,12 @@ void thaw_processes(void)
 	usermodehelper_enable();
 
 	schedule();
+
+#ifdef CONFIG_ARCH_ADVANTECH
+	suspend_key_flag=0;
+	printk( "suspend_key_flag=0.\n");
+#endif
+
 	pr_cont("done.\n");
 	trace_suspend_resume(TPS("thaw_processes"), 0, false);
 }
-- 
1.7.9.5

