Modify i2c switch driver


diff --git a/drivers/i2c/muxes/i2c-mux-pca954x.c b/drivers/i2c/muxes/i2c-mux-pca954x.c
index 550bd36..96bf567 100644
--- a/drivers/i2c/muxes/i2c-mux-pca954x.c
+++ b/drivers/i2c/muxes/i2c-mux-pca954x.c
@@ -295,7 +295,22 @@ static struct i2c_driver pca954x_driver = {
 	.id_table	= pca954x_id,
 };
 
+#ifdef CONFIG_ARCH_ADVANTECH
+static int __init pca954x_init(void)
+{
+	return i2c_add_driver(&pca954x_driver);
+}
+
+static void __exit pca954x_exit(void)
+{
+	i2c_del_driver(&pca954x_driver);
+}
+
+subsys_initcall(pca954x_init);
+module_exit(pca954x_exit);
+#else
 module_i2c_driver(pca954x_driver);
+#endif
 
 MODULE_AUTHOR("Rodolfo Giometti <giometti@linux.it>");
 MODULE_DESCRIPTION("PCA954x I2C mux/switch driver");
-- 
1.7.9.5

