Support EMMC 5.0 in kernel


diff --git a/drivers/mmc/card/block.c b/drivers/mmc/card/block.c
index 8c79235..293ba0d 100644
--- a/drivers/mmc/card/block.c
+++ b/drivers/mmc/card/block.c
@@ -2045,6 +2045,10 @@ static struct mmc_blk_data *mmc_blk_alloc_req(struct mmc_card *card,
 	 * index anymore so we keep track of a name index.
 	 */
 	if (!subname) {
+#ifdef CONFIG_ARCH_ADVANTECH
+		md->name_idx = find_first_zero_bit(name_use, max_devices);
+		printk("MMC: md->name_idx = %d\n", md->name_idx);
+#else
 		int idx;
 
 		idx = mmc_get_reserved_index(card->host);
@@ -2053,7 +2057,7 @@ static struct mmc_blk_data *mmc_blk_alloc_req(struct mmc_card *card,
 		else
 			md->name_idx = find_next_zero_bit(name_use, max_devices,
 					mmc_first_nonreserved_index());
-
+#endif
 		__set_bit(md->name_idx, name_use);
 	} else
 		md->name_idx = ((struct mmc_blk_data *)
-- 
1.7.9.5

