

diff --git a/include/linux/proc-board.h b/include/linux/proc-board.h
index 3f141a7..eb44954 100755
--- a/include/linux/proc-board.h
+++ b/include/linux/proc-board.h
@@ -34,5 +34,8 @@ extern char board_cpu[20];
 #define IS_WISE_3310	( strncmp(board_type,"WISE-3310", 9)==0 ? 1 : 0)
 #define IS_WISE_3310_A1	( strncmp(board_type,"WISE-3310 A1", 12)==0 ? 1 : 0)
 
+#define IS_ROM_7421     ( strncmp(board_type,"ROM-7421", 8)==0 ? 1 : 0)
+#define IS_ROM_7421_A1  ( strncmp(board_type,"ROM-7421 A1", 11)==0 ? 1 : 0)
+
 #endif /* _LINUX_PROC_BOARD_H */
 #endif /* CONFIG_ARCH_ADVANTECH */
-- 
1.7.9.5

