

diff --git a/arch/arm/mach-imx/mach-imx6q.c b/arch/arm/mach-imx/mach-imx6q.c
index b8baa05..38fa26d 100644
--- a/arch/arm/mach-imx/mach-imx6q.c
+++ b/arch/arm/mach-imx/mach-imx6q.c
@@ -209,9 +209,11 @@ static int ar8031_phy_fixup(struct phy_device *dev)
 {
 	u16 val;
 
+#ifndef CONFIG_ARCH_ADVANTECH
 	/* Set RGMII IO voltage to 1.8V */
 	phy_write(dev, 0x1d, 0x1f);
 	phy_write(dev, 0x1e, 0x8);
+#endif
 
 	/* disable phy AR8031 SmartEEE function. */
 	phy_write(dev, 0xd, 0x3);
-- 
1.7.9.5

