From c199a316124f94feb1c7898dd0dec8e8f3710afa Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Mon, 23 May 2016 11:03:35 +0800
Subject: [PATCH] Add to read and write carrier sd card function


diff --git a/drivers/mmc/host/sdhci-esdhc-imx.c b/drivers/mmc/host/sdhci-esdhc-imx.c
old mode 100644
new mode 100755
index b5293f3..d6c05bc
--- a/drivers/mmc/host/sdhci-esdhc-imx.c
+++ b/drivers/mmc/host/sdhci-esdhc-imx.c
@@ -1038,6 +1038,11 @@ sdhci_esdhc_imx_probe_dt(struct platform_device *pdev,
 	struct device_node *np = pdev->dev.of_node;
 	struct esdhc_platform_data *boarddata = &imx_data->boarddata;
 	int ret;
+#ifdef CONFIG_ARCH_ADVANTECH
+	int en_gpios;
+	enum of_gpio_flags en_gpios_flag;
+	int ret1;
+#endif
 
 	if (of_get_property(np, "fsl,wp-controller", NULL))
 		boarddata->wp_type = ESDHC_WP_CONTROLLER;
@@ -1046,6 +1051,19 @@ sdhci_esdhc_imx_probe_dt(struct platform_device *pdev,
 	if (gpio_is_valid(boarddata->wp_gpio))
 		boarddata->wp_type = ESDHC_WP_GPIO;
 
+#ifdef CONFIG_ARCH_ADVANTECH
+	en_gpios = of_get_named_gpio_flags(np, "en-gpios", 0, &en_gpios_flag);
+	if (en_gpios > 0)
+	{
+		ret1 = gpio_request(en_gpios,"en_gpios");
+		if (ret1 < 0) {
+			printk("request en_gpios failed: %d\n", ret1);
+			return ret1;
+		}
+		gpio_direction_output(en_gpios, en_gpios_flag);
+	}
+#endif
+
 	of_property_read_u32(np, "fsl,tuning-step", &boarddata->tuning_step);
 	of_property_read_u32(np, "fsl,tuning-start-tap",
 			     &boarddata->tuning_start_tap);
-- 
1.7.9.5

