

diff --git a/drivers/usb/phy/phy-mxs-usb.c b/drivers/usb/phy/phy-mxs-usb.c
index d5ae0af..5dd080f 100644
--- a/drivers/usb/phy/phy-mxs-usb.c
+++ b/drivers/usb/phy/phy-mxs-usb.c
@@ -25,6 +25,10 @@
 #include <linux/mfd/syscon.h>
 #include <linux/regulator/consumer.h>
 
+#ifdef CONFIG_ARCH_ADVANTECH
+#include <linux/proc-board.h>
+#endif
+
 #define DRIVER_NAME "mxs_phy"
 
 #define HW_USBPHY_PWD				0x00
@@ -240,13 +244,37 @@ static int mxs_phy_hw_init(struct mxs_phy *mxs_phy)
 	if (mxs_phy->data->flags & MXS_PHY_NEED_IP_FIX)
 		writel(BM_USBPHY_IP_FIX, base + HW_USBPHY_IP_SET);
 
-	/* Change D_CAL if necessary */
-	if (mxs_phy->tx_d_cal) {
-		val = readl(base + HW_USBPHY_TX);
-		val &= ~HW_USBPHY_TX_D_CAL_MASK;
-		writel(val | mxs_phy->tx_d_cal, base + HW_USBPHY_TX);
+#ifdef CONFIG_ARCH_ADVANTECH
+	if(mxs_phy->port_id == 0) {
+		/* USB-OTG Port */
+		if ( IS_ROM_3420 || IS_ROM_5420 || IS_ROM_7421) {
+			val = readl(base + HW_USBPHY_TX);
+
+			if (IS_ROM_3420)
+                		val &= 0x10060603;
+			else if (IS_ROM_5420)
+				val &= 0x10060607;
+			else /* IS_ROM_7421 */
+				val &= 0x10000007;
+
+			writel(val, base + HW_USBPHY_TX);
+			udelay(10);
+		}
+		else {
+#endif
+			/* Change D_CAL if necessary */
+			if (mxs_phy->tx_d_cal) {
+			val = readl(base + HW_USBPHY_TX);
+			val &= ~HW_USBPHY_TX_D_CAL_MASK;
+			writel(val | mxs_phy->tx_d_cal, base + HW_USBPHY_TX);
+			}
+#ifdef CONFIG_ARCH_ADVANTECH
+		}
 	}
 
+	printk(KERN_INFO "FREESCALE USB PHY=0x%x\n", readl(base + HW_USBPHY_TX));
+#endif
+
 	return 0;
 }
 
-- 
1.9.1

