From f7b25dfa045c206336c39bb6c0e14954cd8f61f1 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Sat, 19 Mar 2016 20:39:25 +0800
Subject: [PATCH] 4411dts


diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 16497f6..b05ce39 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -275,6 +275,7 @@ dtb-$(CONFIG_ARCH_ADVANTECH) += \
 	imx6dl-ubc220-a1.dtb \
 	imx6q-ubcds31-a1.dtb \
 	imx6q-rom3420-a1.dtb \
+	imx6q-rsb4411-a1.dtb \
 	imx6q-rsb4410-a1.dtb \
 	imx6q-rsb4410-a2.dtb \
 	imx6q-rom5420-a1.dtb \
-- 
1.7.9.5

