From 2d1f90b840bfb80c705690319e51f342ba197e6c Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 24 Mar 2016 00:24:02 +0800
Subject: [PATCH] seconed version


diff --git a/drivers/rtc/rtc-s35390a.c b/drivers/rtc/rtc-s35390a.c
old mode 100644
new mode 100755
index f40afdd..b840490
--- a/drivers/rtc/rtc-s35390a.c
+++ b/drivers/rtc/rtc-s35390a.c
@@ -385,7 +385,7 @@ static int s35390a_probe(struct i2c_client *client,
 	if (s35390a_get_datetime(client, &tm) < 0)
 		dev_warn(&client->dev, "clock needs to be set\n");
 
-	device_set_wakeup_capable(&client->dev, 1);
+	device_init_wakeup(&client->dev, 1);
 
 	s35390a->rtc = devm_rtc_device_register(&client->dev,
 					s35390a_driver.driver.name,
-- 
1.7.9.5

