From 2152c929cbc78393c956f07d18f66f902e88b450 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Wed, 8 Jun 2016 10:28:27 +0800
Subject: [PATCH] Fixed SPI cannot detect issue.


diff --git a/drivers/mtd/devices/m25p80.c b/drivers/mtd/devices/m25p80.c
index 318ff8d..0360410 100644
--- a/drivers/mtd/devices/m25p80.c
+++ b/drivers/mtd/devices/m25p80.c
@@ -262,6 +262,9 @@ static int m25p_remove(struct spi_device *spi)
  * keep them available as module aliases for existing platforms.
  */
 static const struct spi_device_id m25p_ids[] = {
+#ifdef CONFIG_ARCH_ADVANTECH
+	{"n25qba16"},  {"n25qbb16"},
+#endif
 	{"at25fs010"},	{"at25fs040"},	{"at25df041a"},	{"at25df321a"},
 	{"at25df641"},	{"at26f004"},	{"at26df081a"},	{"at26df161a"},
 	{"at26df321"},	{"at45db081d"},
-- 
1.7.9.5

