From 9c18e9e2bd5f4d415635510bf3133d9bbd93a897 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 26 May 2016 14:11:10 +0800
Subject: [PATCH 1/2] 5420solo


diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 9eebe2b..af707ed 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -140,6 +140,7 @@ dtb-$(CONFIG_ARCH_ADVANTECH) += \
 	imx6q-rsb4411-a1.dtb \
 	imx6q-rsb4410-a1.dtb \
 	imx6q-rsb4410-a2.dtb \
+	imx6dl-rom5420-b1.dtb \
 	imx6q-rom5420-a1.dtb \
 	imx6q-rom5420-b1.dtb \
 	imx6q-rom7420-a1.dtb \
-- 
1.7.9.5

