From 2bb62631aaceecf09559e1338564949396b238a3 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Wed, 27 Jul 2016 17:11:45 +0800
Subject: [PATCH 2/3] sata


diff --git a/drivers/ata/ahci_imx.c b/drivers/ata/ahci_imx.c
index da52693..ae7ca6a 100644
--- a/drivers/ata/ahci_imx.c
+++ b/drivers/ata/ahci_imx.c
@@ -291,6 +291,12 @@ static int imx_sata_enable(struct ahci_host_priv *hpriv)
 		sata_gen_phy_reg[1] = 0x059194f6;
 		sata_gen_phy_reg[2] = 0x059194f6;
 	}
+	else if(IS_RSB_4411)
+	{
+		sata_gen_phy_reg[0] = 0x059194f6; 
+		sata_gen_phy_reg[1] = 0x059194f6;
+		sata_gen_phy_reg[2] = 0x059194f6;
+	}
 	else
 	{
 		sata_gen_phy_reg[0] = imxpriv->phy_params; 
-- 
1.7.9.5

