From 59211003171adfef8d5cfbd0be43b6972f45a2ad Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 22 Sep 2016 14:05:09 +0800
Subject: [PATCH] Modify LVDS power on sequence for SI test


diff --git a/drivers/video/backlight/backlight.c b/drivers/video/backlight/backlight.c
index bddc8b1..c250f4d 100644
--- a/drivers/video/backlight/backlight.c
+++ b/drivers/video/backlight/backlight.c
@@ -31,6 +31,11 @@ static const char *const backlight_types[] = {
 	[BACKLIGHT_FIRMWARE] = "firmware",
 };
 
+#if defined(CONFIG_OF) && defined(CONFIG_ARCH_ADVANTECH)
+extern int first_power_on;
+extern void enable_ldb_bkl_pwm(void);
+#endif
+
 #if defined(CONFIG_FB) || (defined(CONFIG_FB_MODULE) && \
 			   defined(CONFIG_BACKLIGHT_CLASS_DEVICE_MODULE))
 /* This callback gets called when something important happens inside a
@@ -61,7 +66,23 @@ static int fb_notifier_callback(struct notifier_block *self,
 				if (!bd->use_count++) {
 					bd->props.state &= ~BL_CORE_FBBLANK;
 					bd->props.fb_blank = FB_BLANK_UNBLANK;
+
+#if defined(CONFIG_OF) && defined(CONFIG_ARCH_ADVANTECH)
+					if(first_power_on)
+						printk(KERN_INFO "[LVDS Sequence] 3 Start to enable LVDS pwm.\n");
+
 					backlight_update_status(bd);
+
+					if(first_power_on)
+					{
+						
+						enable_ldb_bkl_pwm();
+						first_power_on = 0;
+					}
+#else
+					backlight_update_status(bd);
+#endif
+
 				}
 			} else if (fb_blank != FB_BLANK_UNBLANK &&
 				   bd->fb_bl_on[node]) {
diff --git a/drivers/video/backlight/pwm_bl.c b/drivers/video/backlight/pwm_bl.c
index 3af265f..b67bb29 100644
--- a/drivers/video/backlight/pwm_bl.c
+++ b/drivers/video/backlight/pwm_bl.c
@@ -155,17 +155,19 @@ void enable_lcd_vdd_en(void)
 		else
 			gpio_direction_output(lvds_vcc_enable, lvds_vcc_flag);
 	}
+
+	mdelay(250);
 }
 
 void enable_ldb_bkl_pwm(void)
 {
 	int ret;
 
-	mdelay(250);
-
+	mdelay(20);
+	
         if (lvds_bkl_enable > 0)
         {
-		printk(KERN_INFO "[LVDS Sequence] 3 Start to enable LVDS backlight.\n");
+		printk(KERN_INFO "[LVDS Sequence] 4 Start to enable LVDS backlight.\n");
 		ret = gpio_request(lvds_bkl_enable,"lvds_bkl_enable");
 
 		if (ret < 0)
@@ -173,10 +175,6 @@ void enable_ldb_bkl_pwm(void)
 		else
 			gpio_direction_output(lvds_bkl_enable, lvds_bkl_flag);
 	}
-
-	mdelay(30);
-
-	printk(KERN_INFO "[LVDS Sequence] 4 Start to enable LVDS pwm.\n");
 }
 #endif
 
-- 
2.1.2

